{
  "Name": "CurseClient",
  "Process": "Curse.Agent.Host",
  "PID": 8905,
  "Arguments": [
    "/tmp/.mount_CurseFE1hLDJ/resources/app.asar.unpacked/plugins/curse/linux/Curse.Agent.Host",
    "--port=59718",
    "--to-port=59738",
    "--pid=8697",
    "--appDataPath=/home/avery/.config/CurseForge",
    "--clientId=Overwolf"
  ],
  "ProcessStart": "2025-06-12T17:39:53.8489142Z",
  "Version": "1.279.0.26201",
  "BuildDate": "2000-01-01T19:33:22Z",
  "Start": "2025-06-12T13:39:55.856042-04:00",
  "Log": [
{
  "timestamp": "2025-06-12T13:39:55.8362101-04:00",
  "level": "Warn",
  "thread": 1,
  "memory": 0.0,
  "message": "Failed to modified BuiltInConverters: Cannot set initonly static field 'BuiltInConverters' after type 'Newtonsoft.Json.Serialization.DefaultContractResolver' is initialized."
},
{
  "timestamp": "2025-06-12T13:39:55.8500962-04:00",
  "level": "Info",
  "thread": 1,
  "memory": 0.0,
  "message": "Startup Name: CurseForge, Version: 1.279.0.26201"
},
{
  "timestamp": "2025-06-12T13:39:55.8523389-04:00",
  "level": "Info",
  "thread": 1,
  "memory": 0.0,
  "message": "Running Mods Agent with args: --port=59718,--to-port=59738,--pid=8697,--appDataPath=/home/avery/.config/CurseForge,--clientId=Overwolf"
},
{
  "timestamp": "2025-06-12T13:39:55.8618367-04:00",
  "level": "Info",
  "thread": 1,
  "memory": 0.0,
  "message": "start monitor process: 8697"
},
{
  "timestamp": "2025-06-12T13:39:57.969671-04:00",
  "level": "Info",
  "thread": 1,
  "memory": 0.0,
  "message": "Unable to load game instances from disk. No file was found at: /home/avery/.config/CurseForge/agent/GameInstances/ModpackGameInstances.json"
},
{
  "timestamp": "2025-06-12T13:39:58.2156203-04:00",
  "level": "Info",
  "thread": 1,
  "memory": 0.0,
  "message": "Mods Agent server started on port 59718"
},
{
  "timestamp": "2025-06-12T13:39:58.3124693-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Web Socket session connected"
},
{
  "timestamp": "2025-06-12T13:39:58.6716736-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Update simultaneous downloads '1' => '3'"
},
{
  "timestamp": "2025-06-12T13:39:58.6825338-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Initializing Curse Mods Agent"
},
{
  "timestamp": "2025-06-12T13:39:58.6886521-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "update client Info ''..."
},
{
  "timestamp": "2025-06-12T13:39:59.3697039-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Network connected and internet available"
},
{
  "timestamp": "2025-06-12T13:39:59.4314468-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "AddonServiceClient - CfCore Url: https://api.curseforge.com"
},
{
  "timestamp": "2025-06-12T13:39:59.6539092-04:00",
  "level": "Warn",
  "thread": 6,
  "memory": 0.0,
  "message": "Service call failed to: legacy/game/timestamp",
  "error": {
    "message": "Connection refused (api.curseforge.com:443)",
    "type": "System.Net.Http.HttpRequestException",
    "stack": [
      "System.Net.Http.HttpConnectionPool+<ConnectToTcpHostAsync>d__104.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task,ConfigureAwaitOptions)",
      "System.Threading.Tasks.ValueTask`1.get_Result()",
      "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
      "System.Net.Http.HttpConnectionPool+<ConnectAsync>d__103.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task,ConfigureAwaitOptions)",
      "System.Threading.Tasks.ValueTask`1.get_Result()",
      "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
      "System.Net.Http.HttpConnectionPool+<CreateHttp11ConnectionAsync>d__105.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task,ConfigureAwaitOptions)",
      "System.Threading.Tasks.ValueTask`1.get_Result()",
      "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
      "System.Net.Http.HttpConnectionPool+<AddHttp11ConnectionAsync>d__79.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter.GetResult()",
      "System.Threading.Tasks.TaskCompletionSourceWithCancellation`1+<WaitWithCancellationAsync>d__1.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task,ConfigureAwaitOptions)",
      "System.Threading.Tasks.ValueTask`1.get_Result()",
      "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
      "System.Net.Http.HttpConnectionPool+<SendWithVersionDetectionAndRetryAsync>d__89.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task,ConfigureAwaitOptions)",
      "System.Threading.Tasks.ValueTask`1.get_Result()",
      "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
      "System.Net.Http.RedirectHandler+<SendAsync>d__4.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task,ConfigureAwaitOptions)",
      "System.Threading.Tasks.ValueTask`1.get_Result()",
      "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
      "System.Net.Http.DecompressionHandler+<SendAsync>d__16.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter.GetResult()",
      "System.Net.Http.HttpClient+<<SendAsync>g__Core|83_0>d.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()",
      "Curse.Server.Communication.BaseWebServiceClient+<GetAsync>d__14`1.MoveNext()",
      "Curse.Server.Communication.ServicesConfiguration.LogWarning(Exception,String)",
      "Curse.Server.Communication.BaseWebServiceClient+<GetAsync>d__14`1.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishContinuations()",
      "System.Threading.Tasks.Task.FinishStageThree()",
      "System.Threading.Tasks.Task.FinishStageTwo()",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.Finish(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception)",
      "System.Net.Http.HttpClient+<<SendAsync>g__Core|83_0>d.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishContinuations()",
      "System.Threading.Tasks.Task.FinishStageThree()",
      "System.Threading.Tasks.Task.FinishStageTwo()",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.Finish(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncValueTaskMethodBuilder`1.SetException(Exception)",
      "System.Net.Http.DecompressionHandler+<SendAsync>d__16.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishContinuations()",
      "System.Threading.Tasks.Task.FinishStageThree()",
      "System.Threading.Tasks.Task.FinishStageTwo()",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.Finish(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncValueTaskMethodBuilder`1.SetException(Exception)",
      "System.Net.Http.RedirectHandler+<SendAsync>d__4.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishContinuations()",
      "System.Threading.Tasks.Task.FinishStageThree()",
      "System.Threading.Tasks.Task.FinishStageTwo()",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.Finish(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncValueTaskMethodBuilder`1.SetException(Exception)",
      "System.Net.Http.HttpConnectionPool+<SendWithVersionDetectionAndRetryAsync>d__89.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishContinuations()",
      "System.Threading.Tasks.Task.FinishStageThree()",
      "System.Threading.Tasks.Task.FinishStageTwo()",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.Finish(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncValueTaskMethodBuilder`1.SetException(Exception)",
      "System.Threading.Tasks.TaskCompletionSourceWithCancellation`1+<WaitWithCancellationAsync>d__1.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread,ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecuteFromThreadPool(Thread)",
      "System.Threading.ThreadPoolWorkQueue.DispatchWorkItem(Object,Thread)",
      "System.Threading.ThreadPoolWorkQueue.Dispatch()",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerDoWork(PortableThreadPool,Boolean&)",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerThreadStart()",
      "System.Threading.Thread+StartHelper.RunWorker()",
      "System.Threading.Thread+StartHelper.Run()",
      "System.Threading.Thread.StartCallback()"
    ],
    "innerError": {
      "message": "Connection refused",
      "type": "System.Net.Sockets.SocketException",
      "stack": [
        "System.Net.Sockets.Socket+AwaitableSocketAsyncEventArgs.ThrowException(SocketError,CancellationToken)",
        "System.Net.Sockets.Socket+AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource.GetResult(Int16)",
        "System.Threading.Tasks.ValueTask.ThrowIfCompletedUnsuccessfully()",
        "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable+ConfiguredValueTaskAwaiter.GetResult()",
        "System.Net.Sockets.Socket+<<ConnectAsync>g__WaitForConnectWithCancellation|285_0>d.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task,ConfigureAwaitOptions)",
        "System.Threading.Tasks.ValueTask.ThrowIfCompletedUnsuccessfully()",
        "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable+ConfiguredValueTaskAwaiter.GetResult()",
        "System.Net.Http.HttpConnectionPool+<ConnectToTcpHostAsync>d__104.MoveNext()"
      ]
    }
  }
},
{
  "timestamp": "2025-06-12T13:39:59.7494202-04:00",
  "level": "Error",
  "thread": 6,
  "memory": 0.0,
  "message": "Failed getting games database timestamp from server. Request failed with status code InternalServerError"
},
{
  "timestamp": "2025-06-12T13:39:59.7511514-04:00",
  "level": "Error",
  "thread": 6,
  "memory": 0.0,
  "message": "Failed getting games database timestamp from server",
  "error": {
    "message": "Failed to get games timestamp from server",
    "type": "Curse.Server.Communication.Exceptions.ServerRequestException",
    "stack": [
      "Curse.TechService.AddonServiceClient+<GetGameDatabaseTimestampAsync>d__31.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()",
      "Curse.TechService.Games.GameRepository+<GetServiceTimestampAsync>d__14.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishContinuations()",
      "System.Threading.Tasks.Task.FinishStageThree()",
      "System.Threading.Tasks.Task.FinishStageTwo()",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.Finish(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception)",
      "Curse.TechService.AddonServiceClient+<GetGameDatabaseTimestampAsync>d__31.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishContinuations()",
      "System.Threading.Tasks.Task`1.TrySetResult(TResult)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetExistingTaskResult(Task`1,TResult)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetResult(TResult)",
      "Curse.Server.Communication.BaseWebServiceClient+<GetAsync>d__14`1.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishContinuations()",
      "System.Threading.Tasks.Task.FinishStageThree()",
      "System.Threading.Tasks.Task.FinishStageTwo()",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.Finish(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception)",
      "System.Net.Http.HttpClient+<<SendAsync>g__Core|83_0>d.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishContinuations()",
      "System.Threading.Tasks.Task.FinishStageThree()",
      "System.Threading.Tasks.Task.FinishStageTwo()",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.Finish(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncValueTaskMethodBuilder`1.SetException(Exception)",
      "System.Net.Http.DecompressionHandler+<SendAsync>d__16.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishContinuations()",
      "System.Threading.Tasks.Task.FinishStageThree()",
      "System.Threading.Tasks.Task.FinishStageTwo()",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.Finish(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncValueTaskMethodBuilder`1.SetException(Exception)",
      "System.Net.Http.RedirectHandler+<SendAsync>d__4.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishContinuations()",
      "System.Threading.Tasks.Task.FinishStageThree()",
      "System.Threading.Tasks.Task.FinishStageTwo()",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.Finish(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncValueTaskMethodBuilder`1.SetException(Exception)",
      "System.Net.Http.HttpConnectionPool+<SendWithVersionDetectionAndRetryAsync>d__89.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishContinuations()",
      "System.Threading.Tasks.Task.FinishStageThree()",
      "System.Threading.Tasks.Task.FinishStageTwo()",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.Finish(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncValueTaskMethodBuilder`1.SetException(Exception)",
      "System.Threading.Tasks.TaskCompletionSourceWithCancellation`1+<WaitWithCancellationAsync>d__1.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread,ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecuteFromThreadPool(Thread)",
      "System.Threading.ThreadPoolWorkQueue.DispatchWorkItem(Object,Thread)",
      "System.Threading.ThreadPoolWorkQueue.Dispatch()",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerDoWork(PortableThreadPool,Boolean&)",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerThreadStart()",
      "System.Threading.Thread+StartHelper.RunWorker()",
      "System.Threading.Thread+StartHelper.Run()",
      "System.Threading.Thread.StartCallback()"
    ]
  }
},
{
  "timestamp": "2025-06-12T13:39:59.7623825-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Attempt #1 of 2 on GameDatabase -> GetRemoteItems"
},
{
  "timestamp": "2025-06-12T13:39:59.7643577-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Getting Games from server"
},
{
  "timestamp": "2025-06-12T13:39:59.7919866-04:00",
  "level": "Warn",
  "thread": 8,
  "memory": 0.0,
  "message": "Service call failed to: legacy/game?supportsAddons=true",
  "error": {
    "message": "Connection refused (api.curseforge.com:443)",
    "type": "System.Net.Http.HttpRequestException",
    "stack": [
      "System.Net.Http.HttpConnectionPool+<ConnectToTcpHostAsync>d__104.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task,ConfigureAwaitOptions)",
      "System.Threading.Tasks.ValueTask`1.get_Result()",
      "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
      "System.Net.Http.HttpConnectionPool+<ConnectAsync>d__103.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task,ConfigureAwaitOptions)",
      "System.Threading.Tasks.ValueTask`1.get_Result()",
      "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
      "System.Net.Http.HttpConnectionPool+<CreateHttp11ConnectionAsync>d__105.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task,ConfigureAwaitOptions)",
      "System.Threading.Tasks.ValueTask`1.get_Result()",
      "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
      "System.Net.Http.HttpConnectionPool+<AddHttp11ConnectionAsync>d__79.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter.GetResult()",
      "System.Threading.Tasks.TaskCompletionSourceWithCancellation`1+<WaitWithCancellationAsync>d__1.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task,ConfigureAwaitOptions)",
      "System.Threading.Tasks.ValueTask`1.get_Result()",
      "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
      "System.Net.Http.HttpConnectionPool+<SendWithVersionDetectionAndRetryAsync>d__89.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task,ConfigureAwaitOptions)",
      "System.Threading.Tasks.ValueTask`1.get_Result()",
      "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
      "System.Net.Http.RedirectHandler+<SendAsync>d__4.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task,ConfigureAwaitOptions)",
      "System.Threading.Tasks.ValueTask`1.get_Result()",
      "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
      "System.Net.Http.DecompressionHandler+<SendAsync>d__16.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter.GetResult()",
      "System.Net.Http.HttpClient+<<SendAsync>g__Core|83_0>d.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()",
      "Curse.Server.Communication.BaseWebServiceClient+<GetAsync>d__14`1.MoveNext()",
      "Curse.Server.Communication.ServicesConfiguration.LogWarning(Exception,String)",
      "Curse.Server.Communication.BaseWebServiceClient+<GetAsync>d__14`1.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishContinuations()",
      "System.Threading.Tasks.Task.FinishStageThree()",
      "System.Threading.Tasks.Task.FinishStageTwo()",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.Finish(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception)",
      "System.Net.Http.HttpClient+<<SendAsync>g__Core|83_0>d.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishContinuations()",
      "System.Threading.Tasks.Task.FinishStageThree()",
      "System.Threading.Tasks.Task.FinishStageTwo()",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.Finish(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncValueTaskMethodBuilder`1.SetException(Exception)",
      "System.Net.Http.DecompressionHandler+<SendAsync>d__16.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishContinuations()",
      "System.Threading.Tasks.Task.FinishStageThree()",
      "System.Threading.Tasks.Task.FinishStageTwo()",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.Finish(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncValueTaskMethodBuilder`1.SetException(Exception)",
      "System.Net.Http.RedirectHandler+<SendAsync>d__4.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishContinuations()",
      "System.Threading.Tasks.Task.FinishStageThree()",
      "System.Threading.Tasks.Task.FinishStageTwo()",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.Finish(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncValueTaskMethodBuilder`1.SetException(Exception)",
      "System.Net.Http.HttpConnectionPool+<SendWithVersionDetectionAndRetryAsync>d__89.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishContinuations()",
      "System.Threading.Tasks.Task.FinishStageThree()",
      "System.Threading.Tasks.Task.FinishStageTwo()",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.Finish(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncValueTaskMethodBuilder`1.SetException(Exception)",
      "System.Threading.Tasks.TaskCompletionSourceWithCancellation`1+<WaitWithCancellationAsync>d__1.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread,ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecuteFromThreadPool(Thread)",
      "System.Threading.ThreadPoolWorkQueue.DispatchWorkItem(Object,Thread)",
      "System.Threading.ThreadPoolWorkQueue.Dispatch()",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerDoWork(PortableThreadPool,Boolean&)",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerThreadStart()",
      "System.Threading.Thread+StartHelper.RunWorker()",
      "System.Threading.Thread+StartHelper.Run()",
      "System.Threading.Thread.StartCallback()"
    ],
    "innerError": {
      "message": "Connection refused",
      "type": "System.Net.Sockets.SocketException",
      "stack": [
        "System.Net.Sockets.Socket+AwaitableSocketAsyncEventArgs.ThrowException(SocketError,CancellationToken)",
        "System.Net.Sockets.Socket+AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource.GetResult(Int16)",
        "System.Threading.Tasks.ValueTask.ThrowIfCompletedUnsuccessfully()",
        "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable+ConfiguredValueTaskAwaiter.GetResult()",
        "System.Net.Sockets.Socket+<<ConnectAsync>g__WaitForConnectWithCancellation|285_0>d.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task,ConfigureAwaitOptions)",
        "System.Threading.Tasks.ValueTask.ThrowIfCompletedUnsuccessfully()",
        "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable+ConfiguredValueTaskAwaiter.GetResult()",
        "System.Net.Http.HttpConnectionPool+<ConnectToTcpHostAsync>d__104.MoveNext()"
      ]
    }
  }
},
{
  "timestamp": "2025-06-12T13:39:59.7955721-04:00",
  "level": "Error",
  "thread": 8,
  "memory": 0.0,
  "message": "Failed getting games from server. Request failed with status code InternalServerError"
},
{
  "timestamp": "2025-06-12T13:40:04.7985577-04:00",
  "level": "Info",
  "thread": 13,
  "memory": 0.0,
  "message": "Attempt #2 of 2 on GameDatabase -> GetRemoteItems"
},
{
  "timestamp": "2025-06-12T13:40:04.7988403-04:00",
  "level": "Info",
  "thread": 13,
  "memory": 0.0,
  "message": "Getting Games from server"
},
{
  "timestamp": "2025-06-12T13:40:04.82359-04:00",
  "level": "Warn",
  "thread": 6,
  "memory": 0.0,
  "message": "Service call failed to: legacy/game?supportsAddons=true",
  "error": {
    "message": "Connection refused (api.curseforge.com:443)",
    "type": "System.Net.Http.HttpRequestException",
    "stack": [
      "System.Net.Http.HttpConnectionPool+<ConnectToTcpHostAsync>d__104.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Threading.Tasks.ValueTask`1.get_Result()",
      "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
      "System.Net.Http.HttpConnectionPool+<ConnectAsync>d__103.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Threading.Tasks.ValueTask`1.get_Result()",
      "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
      "System.Net.Http.HttpConnectionPool+<CreateHttp11ConnectionAsync>d__105.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Threading.Tasks.ValueTask`1.get_Result()",
      "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
      "System.Net.Http.HttpConnectionPool+<AddHttp11ConnectionAsync>d__79.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter.GetResult()",
      "System.Threading.Tasks.TaskCompletionSourceWithCancellation`1+<WaitWithCancellationAsync>d__1.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Threading.Tasks.ValueTask`1.get_Result()",
      "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
      "System.Net.Http.HttpConnectionPool+<SendWithVersionDetectionAndRetryAsync>d__89.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Threading.Tasks.ValueTask`1.get_Result()",
      "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
      "System.Net.Http.RedirectHandler+<SendAsync>d__4.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Threading.Tasks.ValueTask`1.get_Result()",
      "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
      "System.Net.Http.DecompressionHandler+<SendAsync>d__16.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter.GetResult()",
      "System.Net.Http.HttpClient+<<SendAsync>g__Core|83_0>d.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()",
      "Curse.Server.Communication.BaseWebServiceClient+<GetAsync>d__14`1.MoveNext()",
      "Curse.Server.Communication.ServicesConfiguration.LogWarning(Exception,String)",
      "Curse.Server.Communication.BaseWebServiceClient+<GetAsync>d__14`1.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishStageThree()",
      "System.Threading.Tasks.Task.FinishStageTwo()",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.Finish(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception)",
      "System.Net.Http.HttpClient+<<SendAsync>g__Core|83_0>d.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishStageThree()",
      "System.Threading.Tasks.Task.FinishStageTwo()",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.Finish(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncValueTaskMethodBuilder`1.SetException(Exception)",
      "System.Net.Http.DecompressionHandler+<SendAsync>d__16.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishStageThree()",
      "System.Threading.Tasks.Task.FinishStageTwo()",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.Finish(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncValueTaskMethodBuilder`1.SetException(Exception)",
      "System.Net.Http.RedirectHandler+<SendAsync>d__4.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishStageThree()",
      "System.Threading.Tasks.Task.FinishStageTwo()",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.Finish(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncValueTaskMethodBuilder`1.SetException(Exception)",
      "System.Net.Http.HttpConnectionPool+<SendWithVersionDetectionAndRetryAsync>d__89.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishStageThree()",
      "System.Threading.Tasks.Task.FinishStageTwo()",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.Finish(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncValueTaskMethodBuilder`1.SetException(Exception)",
      "System.Threading.Tasks.TaskCompletionSourceWithCancellation`1+<WaitWithCancellationAsync>d__1.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread,ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecuteFromThreadPool(Thread)",
      "System.Threading.ThreadPoolWorkQueue.Dispatch()",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerDoWork(PortableThreadPool,Boolean&)",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerThreadStart()",
      "System.Threading.Thread+StartHelper.RunWorker()",
      "System.Threading.Thread+StartHelper.Run()",
      "System.Threading.Thread.StartCallback()"
    ],
    "innerError": {
      "message": "Connection refused",
      "type": "System.Net.Sockets.SocketException",
      "stack": [
        "System.Net.Sockets.Socket+AwaitableSocketAsyncEventArgs.ThrowException(SocketError,CancellationToken)",
        "System.Net.Sockets.Socket+AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource.GetResult(Int16)",
        "System.Threading.Tasks.ValueTask.ThrowIfCompletedUnsuccessfully()",
        "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable+ConfiguredValueTaskAwaiter.GetResult()",
        "System.Net.Sockets.Socket+<<ConnectAsync>g__WaitForConnectWithCancellation|285_0>d.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Threading.Tasks.ValueTask.ThrowIfCompletedUnsuccessfully()",
        "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable+ConfiguredValueTaskAwaiter.GetResult()",
        "System.Net.Http.HttpConnectionPool+<ConnectToTcpHostAsync>d__104.MoveNext()"
      ]
    }
  }
},
{
  "timestamp": "2025-06-12T13:40:04.8381152-04:00",
  "level": "Error",
  "thread": 6,
  "memory": 0.0,
  "message": "Failed getting games from server. Request failed with status code InternalServerError"
},
{
  "timestamp": "2025-06-12T13:40:14.8420187-04:00",
  "level": "Error",
  "thread": 6,
  "memory": 0.0,
  "message": "Failed getting games from server",
  "error": {
    "message": "One or more errors occurred. (Failed to get games from server) (Failed to get games from server)",
    "type": "System.AggregateException",
    "stack": [
      "Curse.Radium.Utilities.Retry+<DoAsync>d__5`1.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()",
      "Curse.TechService.Games.GameRepository+<GetRemoteItemsAsync>d__16.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishStageThree()",
      "System.Threading.Tasks.Task.FinishStageTwo()",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.Finish(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception)",
      "Curse.Radium.Utilities.Retry+<DoAsync>d__5`1.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task+DelayPromise.CompleteTimedOut()",
      "System.Threading.TimerQueueTimer.Fire(Boolean)",
      "System.Threading.TimerQueue.FireNextTimers()",
      "System.Threading.ThreadPoolWorkQueue.Dispatch()",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerDoWork(PortableThreadPool,Boolean&)",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerThreadStart()",
      "System.Threading.Thread+StartHelper.RunWorker()",
      "System.Threading.Thread+StartHelper.Run()",
      "System.Threading.Thread.StartCallback()"
    ],
    "innerError": {
      "message": "Failed to get games from server",
      "type": "Curse.Server.Communication.Exceptions.ServerRequestException",
      "stack": [
        "Curse.TechService.AddonServiceClient+<GetGamesAsync>d__30.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task,ConfigureAwaitOptions)",
        "System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()",
        "Curse.TechService.Games.GameRepository+<<GetRemoteItemsAsync>b__16_0>d.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task,ConfigureAwaitOptions)",
        "System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()",
        "Curse.Radium.Utilities.Retry+<DoAsync>d__5`1.MoveNext()"
      ]
    }
  }
},
{
  "timestamp": "2025-06-12T13:40:14.8532789-04:00",
  "level": "Warn",
  "thread": 6,
  "memory": 0.0,
  "message": "Failed getting games list from server. Attempting to load from local cache"
},
{
  "timestamp": "2025-06-12T13:40:14.8542001-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Loading games from local cache at /home/avery/.config/CurseForge/agent/gamesCache.json"
},
{
  "timestamp": "2025-06-12T13:40:15.1395213-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Loaded 25 games from the local games cache"
},
{
  "timestamp": "2025-06-12T13:40:15.160691-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "[GameRepository] Completed download of cache",
  "data": {
    "name": "Games",
    "elapsedTime": "15,382.00"
  }
},
{
  "timestamp": "2025-06-12T13:40:15.2099287-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Initializing local databases..."
},
{
  "timestamp": "2025-06-12T13:40:15.2107853-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Initializing game instances."
},
{
  "timestamp": "2025-06-12T13:40:15.2511272-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Found RegisterType on Type: 'Curse.Radium.Addons.AddonGameInstance'"
},
{
  "timestamp": "2025-06-12T13:40:15.2522184-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Found RegisterType on Type: 'Curse.Games.Minecraft.MinecraftGameInstance'"
},
{
  "timestamp": "2025-06-12T13:40:15.2529327-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Initializing game instance of type Curse.Radium.Addons.AddonGameInstance"
},
{
  "timestamp": "2025-06-12T13:40:15.3506474-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "[AddonGameInstance] LoadInstancesFromDisk loaded 0 instances"
},
{
  "timestamp": "2025-06-12T13:40:15.3509027-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Initializing game instance of type Curse.Games.Minecraft.MinecraftGameInstance"
},
{
  "timestamp": "2025-06-12T13:40:16.1560295-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "[MinecraftGameInstance] LoadInstancesFromDisk loaded 3 instances"
},
{
  "timestamp": "2025-06-12T13:40:16.170508-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Game Minecraft (Id: 432) was detected at /home/avery/Documents/curseforge/minecraft/Instances/fun/"
},
{
  "timestamp": "2025-06-12T13:40:16.2410889-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Initialized game fun instance"
},
{
  "timestamp": "2025-06-12T13:40:16.2412473-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Game Minecraft (Id: 432) was detected at /home/avery/Documents/curseforge/minecraft/Instances/e/"
},
{
  "timestamp": "2025-06-12T13:40:16.2610134-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Initialized game e instance"
},
{
  "timestamp": "2025-06-12T13:40:16.2610761-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Game Minecraft (Id: 432) was detected at /home/avery/Documents/curseforge/minecraft/Instances/Homestead - A Cozy Survival Experience/"
},
{
  "timestamp": "2025-06-12T13:40:16.2905187-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Initialized game Homestead - A Cozy Survival Experience instance"
},
{
  "timestamp": "2025-06-12T13:40:16.3039465-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Game instances initialization completed. Operation took 1085 ms"
},
{
  "timestamp": "2025-06-12T13:40:16.3077581-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Initializing Modpack instances."
},
{
  "timestamp": "2025-06-12T13:40:16.3249391-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Attempt #1 of 3 on InstalledAddonDatabase -> GetAddons"
},
{
  "timestamp": "2025-06-12T13:40:21.3790048-04:00",
  "level": "Info",
  "thread": 8,
  "memory": 0.0,
  "message": "Attempt #2 of 3 on InstalledAddonDatabase -> GetAddons"
},
{
  "timestamp": "2025-06-12T13:40:31.4089385-04:00",
  "level": "Info",
  "thread": 14,
  "memory": 0.0,
  "message": "Attempt #3 of 3 on InstalledAddonDatabase -> GetAddons"
},
{
  "timestamp": "2025-06-12T13:40:41.4346832-04:00",
  "level": "Error",
  "thread": 14,
  "memory": 0.0,
  "message": "UpdateCache Exception!",
  "error": {
    "message": "One or more errors occurred. (Connection refused (api.curseforge.com:443)) (Connection refused (api.curseforge.com:443)) (Connection refused (api.curseforge.com:443))",
    "type": "System.AggregateException",
    "stack": [
      "Curse.Radium.Utilities.Retry+<DoAsync>d__5`1.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter.GetResult()",
      "Curse.Radium.Addons.ProjectRepository+<UpdateProjectsAsync>d__22.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception)",
      "Curse.Radium.Utilities.Retry+<DoAsync>d__5`1.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task+DelayPromise.CompleteTimedOut()",
      "System.Threading.TimerQueueTimer.Fire(Boolean)",
      "System.Threading.TimerQueue.FireNextTimers()",
      "System.Threading.ThreadPoolWorkQueue.Dispatch()",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerDoWork(PortableThreadPool,Boolean&)",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerThreadStart()",
      "System.Threading.Thread+StartHelper.RunWorker()",
      "System.Threading.Thread+StartHelper.Run()",
      "System.Threading.Thread.StartCallback()"
    ],
    "innerError": {
      "message": "Connection refused (api.curseforge.com:443)",
      "type": "System.Net.Http.HttpRequestException",
      "stack": [
        "System.Net.Http.HttpConnectionPool+<ConnectToTcpHostAsync>d__104.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Threading.Tasks.ValueTask`1.get_Result()",
        "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
        "System.Net.Http.HttpConnectionPool+<ConnectAsync>d__103.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Threading.Tasks.ValueTask`1.get_Result()",
        "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
        "System.Net.Http.HttpConnectionPool+<CreateHttp11ConnectionAsync>d__105.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Threading.Tasks.ValueTask`1.get_Result()",
        "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
        "System.Net.Http.HttpConnectionPool+<AddHttp11ConnectionAsync>d__79.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter.GetResult()",
        "System.Threading.Tasks.TaskCompletionSourceWithCancellation`1+<WaitWithCancellationAsync>d__1.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Threading.Tasks.ValueTask`1.get_Result()",
        "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
        "System.Net.Http.HttpConnectionPool+<SendWithVersionDetectionAndRetryAsync>d__89.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Threading.Tasks.ValueTask`1.get_Result()",
        "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
        "System.Net.Http.RedirectHandler+<SendAsync>d__4.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Threading.Tasks.ValueTask`1.get_Result()",
        "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
        "System.Net.Http.DecompressionHandler+<SendAsync>d__16.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter.GetResult()",
        "System.Net.Http.HttpClient+<<SendAsync>g__Core|83_0>d.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()",
        "Curse.Server.Communication.BaseWebServiceClient+<PostAsync>d__15`1.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()",
        "Curse.TechService.AddonServiceClient+<GetProjectsAsync>d__11.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()",
        "Curse.Radium.Addons.ProjectRepository+<>c__DisplayClass22_0+<<UpdateProjectsAsync>b__0>d.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Runtime.CompilerServices.TaskAwaiter.GetResult()",
        "Curse.Radium.Utilities.Retry+<>c__DisplayClass4_0+<<DoAsync>b__0>d.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()",
        "Curse.Radium.Utilities.Retry+<DoAsync>d__5`1.MoveNext()"
      ],
      "innerError": {
        "message": "Connection refused",
        "type": "System.Net.Sockets.SocketException",
        "stack": [
          "System.Net.Sockets.Socket+AwaitableSocketAsyncEventArgs.ThrowException(SocketError,CancellationToken)",
          "System.Net.Sockets.Socket+AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource.GetResult(Int16)",
          "System.Threading.Tasks.ValueTask.ThrowIfCompletedUnsuccessfully()",
          "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable+ConfiguredValueTaskAwaiter.GetResult()",
          "System.Net.Sockets.Socket+<<ConnectAsync>g__WaitForConnectWithCancellation|285_0>d.MoveNext()",
          "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
          "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
          "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
          "System.Threading.Tasks.ValueTask.ThrowIfCompletedUnsuccessfully()",
          "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable+ConfiguredValueTaskAwaiter.GetResult()",
          "System.Net.Http.HttpConnectionPool+<ConnectToTcpHostAsync>d__104.MoveNext()"
        ]
      }
    }
  }
},
{
  "timestamp": "2025-06-12T13:40:41.4782211-04:00",
  "level": "Info",
  "thread": 14,
  "memory": 0.0,
  "message": "Running startup game detection"
},
{
  "timestamp": "2025-06-12T13:40:41.5151314-04:00",
  "level": "Info",
  "thread": 14,
  "memory": 0.0,
  "message": "Performing quick game detection for 25 games..."
},
{
  "timestamp": "2025-06-12T13:40:41.5752139-04:00",
  "level": "Info",
  "thread": 14,
  "memory": 0.0,
  "message": "Completed startup game detection. We have a total of 3 detected game instances. Operation took 96 ms"
},
{
  "timestamp": "2025-06-12T13:40:41.5880587-04:00",
  "level": "Info",
  "thread": 14,
  "memory": 0.0,
  "message": "Validating Minecraft installation"
},
{
  "timestamp": "2025-06-12T13:40:41.5929043-04:00",
  "level": "Info",
  "thread": 14,
  "memory": 0.0,
  "message": "/home/avery/Documents/curseforge/minecraft/Install/minecraft-launcher found."
},
{
  "timestamp": "2025-06-12T13:40:41.7263577-04:00",
  "level": "Info",
  "thread": 14,
  "memory": 0.0,
  "message": "Setting file watcher on instance directory /home/avery/Documents/curseforge/minecraft/Instances"
},
{
  "timestamp": "2025-06-12T13:40:41.7615091-04:00",
  "level": "Warn",
  "thread": 8,
  "memory": 0.0,
  "message": "Service call failed to: v1/minecraft/modloader/timestamp",
  "error": {
    "message": "Connection refused (api.curseforge.com:443)",
    "type": "System.Net.Http.HttpRequestException",
    "stack": [
      "System.Net.Http.HttpConnectionPool+<ConnectToTcpHostAsync>d__104.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Threading.Tasks.ValueTask`1.get_Result()",
      "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
      "System.Net.Http.HttpConnectionPool+<ConnectAsync>d__103.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Threading.Tasks.ValueTask`1.get_Result()",
      "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
      "System.Net.Http.HttpConnectionPool+<CreateHttp11ConnectionAsync>d__105.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Threading.Tasks.ValueTask`1.get_Result()",
      "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
      "System.Net.Http.HttpConnectionPool+<AddHttp11ConnectionAsync>d__79.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter.GetResult()",
      "System.Threading.Tasks.TaskCompletionSourceWithCancellation`1+<WaitWithCancellationAsync>d__1.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Threading.Tasks.ValueTask`1.get_Result()",
      "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
      "System.Net.Http.HttpConnectionPool+<SendWithVersionDetectionAndRetryAsync>d__89.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Threading.Tasks.ValueTask`1.get_Result()",
      "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
      "System.Net.Http.RedirectHandler+<SendAsync>d__4.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Threading.Tasks.ValueTask`1.get_Result()",
      "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
      "System.Net.Http.DecompressionHandler+<SendAsync>d__16.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter.GetResult()",
      "System.Net.Http.HttpClient+<<SendAsync>g__Core|83_0>d.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()",
      "Curse.Server.Communication.BaseWebServiceClient+<GetAsync>d__14`1.MoveNext()",
      "Curse.Server.Communication.ServicesConfiguration.LogWarning(Exception,String)",
      "Curse.Server.Communication.BaseWebServiceClient+<GetAsync>d__14`1.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception)",
      "System.Net.Http.HttpClient+<<SendAsync>g__Core|83_0>d.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncValueTaskMethodBuilder`1.SetException(Exception)",
      "System.Net.Http.DecompressionHandler+<SendAsync>d__16.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncValueTaskMethodBuilder`1.SetException(Exception)",
      "System.Net.Http.RedirectHandler+<SendAsync>d__4.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncValueTaskMethodBuilder`1.SetException(Exception)",
      "System.Net.Http.HttpConnectionPool+<SendWithVersionDetectionAndRetryAsync>d__89.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncValueTaskMethodBuilder`1.SetException(Exception)",
      "System.Threading.Tasks.TaskCompletionSourceWithCancellation`1+<WaitWithCancellationAsync>d__1.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread,ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecuteFromThreadPool(Thread)",
      "System.Threading.ThreadPoolWorkQueue.Dispatch()",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerDoWork(PortableThreadPool,Boolean&)",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerThreadStart()",
      "System.Threading.Thread+StartHelper.RunWorker()",
      "System.Threading.Thread+StartHelper.Run()",
      "System.Threading.Thread.StartCallback()"
    ],
    "innerError": {
      "message": "Connection refused",
      "type": "System.Net.Sockets.SocketException",
      "stack": [
        "System.Net.Sockets.Socket+AwaitableSocketAsyncEventArgs.ThrowException(SocketError,CancellationToken)",
        "System.Net.Sockets.Socket+AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource.GetResult(Int16)",
        "System.Threading.Tasks.ValueTask.ThrowIfCompletedUnsuccessfully()",
        "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable+ConfiguredValueTaskAwaiter.GetResult()",
        "System.Net.Sockets.Socket+<<ConnectAsync>g__WaitForConnectWithCancellation|285_0>d.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Threading.Tasks.ValueTask.ThrowIfCompletedUnsuccessfully()",
        "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable+ConfiguredValueTaskAwaiter.GetResult()",
        "System.Net.Http.HttpConnectionPool+<ConnectToTcpHostAsync>d__104.MoveNext()"
      ]
    }
  }
},
{
  "timestamp": "2025-06-12T13:40:41.7693296-04:00",
  "level": "Error",
  "thread": 8,
  "memory": 0.0,
  "message": "Failed getting modloaders database timestamp from server. Request failed with status code InternalServerError"
},
{
  "timestamp": "2025-06-12T13:40:41.7712528-04:00",
  "level": "Error",
  "thread": 8,
  "memory": 0.0,
  "message": "Failed getting modloaders database timestamp from server",
  "error": {
    "message": "Failed to get modloaders timestamp from server",
    "type": "Curse.Server.Communication.Exceptions.ServerRequestException",
    "stack": [
      "Curse.TechService.AddonServiceClient+<GetModloadersDatabaseTimestampAsync>d__27.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()",
      "Curse.Games.Minecraft.ModLoader.ModLoaderRepository+<GetServiceTimestampAsync>d__17.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception)",
      "Curse.TechService.AddonServiceClient+<GetModloadersDatabaseTimestampAsync>d__27.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task`1.TrySetResult(TResult)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetExistingTaskResult(Task`1,TResult)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetResult(TResult)",
      "Curse.Server.Communication.BaseWebServiceClient+<GetAsync>d__14`1.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception)",
      "System.Net.Http.HttpClient+<<SendAsync>g__Core|83_0>d.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncValueTaskMethodBuilder`1.SetException(Exception)",
      "System.Net.Http.DecompressionHandler+<SendAsync>d__16.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncValueTaskMethodBuilder`1.SetException(Exception)",
      "System.Net.Http.RedirectHandler+<SendAsync>d__4.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncValueTaskMethodBuilder`1.SetException(Exception)",
      "System.Net.Http.HttpConnectionPool+<SendWithVersionDetectionAndRetryAsync>d__89.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncValueTaskMethodBuilder`1.SetException(Exception)",
      "System.Threading.Tasks.TaskCompletionSourceWithCancellation`1+<WaitWithCancellationAsync>d__1.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread,ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecuteFromThreadPool(Thread)",
      "System.Threading.ThreadPoolWorkQueue.Dispatch()",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerDoWork(PortableThreadPool,Boolean&)",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerThreadStart()",
      "System.Threading.Thread+StartHelper.RunWorker()",
      "System.Threading.Thread+StartHelper.Run()",
      "System.Threading.Thread.StartCallback()"
    ]
  }
},
{
  "timestamp": "2025-06-12T13:40:41.8126785-04:00",
  "level": "Warn",
  "thread": 8,
  "memory": 0.0,
  "message": "Service call failed to: v1/minecraft/modloader?includeAll=true",
  "error": {
    "message": "Connection refused (api.curseforge.com:443)",
    "type": "System.Net.Http.HttpRequestException",
    "stack": [
      "System.Net.Http.HttpConnectionPool+<ConnectToTcpHostAsync>d__104.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Threading.Tasks.ValueTask`1.get_Result()",
      "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
      "System.Net.Http.HttpConnectionPool+<ConnectAsync>d__103.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Threading.Tasks.ValueTask`1.get_Result()",
      "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
      "System.Net.Http.HttpConnectionPool+<CreateHttp11ConnectionAsync>d__105.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Threading.Tasks.ValueTask`1.get_Result()",
      "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
      "System.Net.Http.HttpConnectionPool+<AddHttp11ConnectionAsync>d__79.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter.GetResult()",
      "System.Threading.Tasks.TaskCompletionSourceWithCancellation`1+<WaitWithCancellationAsync>d__1.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Threading.Tasks.ValueTask`1.get_Result()",
      "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
      "System.Net.Http.HttpConnectionPool+<SendWithVersionDetectionAndRetryAsync>d__89.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Threading.Tasks.ValueTask`1.get_Result()",
      "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
      "System.Net.Http.RedirectHandler+<SendAsync>d__4.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Threading.Tasks.ValueTask`1.get_Result()",
      "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
      "System.Net.Http.DecompressionHandler+<SendAsync>d__16.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter.GetResult()",
      "System.Net.Http.HttpClient+<<SendAsync>g__Core|83_0>d.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()",
      "Curse.Server.Communication.BaseWebServiceClient+<GetAsync>d__14`1.MoveNext()",
      "Curse.Server.Communication.ServicesConfiguration.LogWarning(Exception,String)",
      "Curse.Server.Communication.BaseWebServiceClient+<GetAsync>d__14`1.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception)",
      "System.Net.Http.HttpClient+<<SendAsync>g__Core|83_0>d.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncValueTaskMethodBuilder`1.SetException(Exception)",
      "System.Net.Http.DecompressionHandler+<SendAsync>d__16.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncValueTaskMethodBuilder`1.SetException(Exception)",
      "System.Net.Http.RedirectHandler+<SendAsync>d__4.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncValueTaskMethodBuilder`1.SetException(Exception)",
      "System.Net.Http.HttpConnectionPool+<SendWithVersionDetectionAndRetryAsync>d__89.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncValueTaskMethodBuilder`1.SetException(Exception)",
      "System.Threading.Tasks.TaskCompletionSourceWithCancellation`1+<WaitWithCancellationAsync>d__1.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread,ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecuteFromThreadPool(Thread)",
      "System.Threading.ThreadPoolWorkQueue.Dispatch()",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerDoWork(PortableThreadPool,Boolean&)",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerThreadStart()",
      "System.Threading.Thread+StartHelper.RunWorker()",
      "System.Threading.Thread+StartHelper.Run()",
      "System.Threading.Thread.StartCallback()"
    ],
    "innerError": {
      "message": "Connection refused",
      "type": "System.Net.Sockets.SocketException",
      "stack": [
        "System.Net.Sockets.Socket+AwaitableSocketAsyncEventArgs.ThrowException(SocketError,CancellationToken)",
        "System.Net.Sockets.Socket+AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource.GetResult(Int16)",
        "System.Threading.Tasks.ValueTask.ThrowIfCompletedUnsuccessfully()",
        "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable+ConfiguredValueTaskAwaiter.GetResult()",
        "System.Net.Sockets.Socket+<<ConnectAsync>g__WaitForConnectWithCancellation|285_0>d.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Threading.Tasks.ValueTask.ThrowIfCompletedUnsuccessfully()",
        "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable+ConfiguredValueTaskAwaiter.GetResult()",
        "System.Net.Http.HttpConnectionPool+<ConnectToTcpHostAsync>d__104.MoveNext()"
      ]
    }
  }
},
{
  "timestamp": "2025-06-12T13:40:41.8203476-04:00",
  "level": "Error",
  "thread": 8,
  "memory": 0.0,
  "message": "Failed getting Minecraft Modloaders from server. Request failed with status code InternalServerError"
},
{
  "timestamp": "2025-06-12T13:40:41.8324912-04:00",
  "level": "Info",
  "thread": 18,
  "memory": 0.0,
  "message": "Refreshing Minecraft instances list"
},
{
  "timestamp": "2025-06-12T13:40:41.8330763-04:00",
  "level": "Info",
  "thread": 14,
  "memory": 0.0,
  "message": "Initializing tasks"
},
{
  "timestamp": "2025-06-12T13:40:41.8476797-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Resolving Minecraft instances list from /home/avery/Documents/curseforge/minecraft/Instances"
},
{
  "timestamp": "2025-06-12T13:40:41.8576551-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Skipping existing minecraft game instance at /home/avery/Documents/curseforge/minecraft/Instances/e"
},
{
  "timestamp": "2025-06-12T13:40:41.8579699-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Skipping existing minecraft game instance at /home/avery/Documents/curseforge/minecraft/Instances/Homestead - A Cozy Survival Experience"
},
{
  "timestamp": "2025-06-12T13:40:41.858098-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Skipping existing minecraft game instance at /home/avery/Documents/curseforge/minecraft/Instances/fun"
},
{
  "timestamp": "2025-06-12T13:40:41.869786-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Attempt #1 of 3 on InstalledAddonDatabase -> GetAddons"
},
{
  "timestamp": "2025-06-12T13:40:41.8955366-04:00",
  "level": "Info",
  "thread": 14,
  "memory": 0.0,
  "message": "Discord RPC client initialized successfully."
},
{
  "timestamp": "2025-06-12T13:40:41.8972227-04:00",
  "level": "Info",
  "thread": 14,
  "memory": 0.0,
  "message": "Initializing Periodic Tasks"
},
{
  "timestamp": "2025-06-12T13:40:41.8965785-04:00",
  "level": "Error",
  "thread": 2,
  "memory": 0.0,
  "message": "Global Unhandled Exception",
  "data": {
    "Exception": {
      "Type": "System.AggregateException",
      "Message": "A Task's exception(s) were not observed either by Waiting on the Task or accessing its Exception property. As a result, the unobserved exception was rethrown by the finalizer thread. (Failed to get Minecraft Modloaders from server)",
      "Inner": {
        "Type": "Curse.Server.Communication.Exceptions.ServerRequestException",
        "Message": "Failed to get Minecraft Modloaders from server",
        "Stack": [
          "Curse.TechService.AddonServiceClient+<GetModloadersAsync>d__26.MoveNext()",
          "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
          "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
          "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
          "System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()",
          "Curse.Games.Minecraft.ModLoader.ModLoaderRepository+<GetRemoteItemsAsync>d__21.MoveNext()",
          "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
          "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
          "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
          "System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()",
          "Curse.Radium.Addons.BaseAddonServiceDatabase`3+<UpdateCacheAsync>d__14.MoveNext()",
          "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
          "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
          "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
          "System.Runtime.CompilerServices.TaskAwaiter.GetResult()",
          "Curse.Radium.Addons.BaseAddonServiceDatabase`3+<InitializeAsync>d__8.MoveNext()",
          "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
          "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
          "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
          "System.Runtime.CompilerServices.TaskAwaiter.GetResult()",
          "Curse.AppService.MinecraftAppService+<<InitializeAsync>b__22_0>d.MoveNext()"
        ]
      }
    }
  },
  "error": {
    "message": "A Task's exception(s) were not observed either by Waiting on the Task or accessing its Exception property. As a result, the unobserved exception was rethrown by the finalizer thread. (Failed to get Minecraft Modloaders from server)",
    "type": "System.AggregateException",
    "innerError": {
      "message": "Failed to get Minecraft Modloaders from server",
      "type": "Curse.Server.Communication.Exceptions.ServerRequestException",
      "stack": [
        "Curse.TechService.AddonServiceClient+<GetModloadersAsync>d__26.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()",
        "Curse.Games.Minecraft.ModLoader.ModLoaderRepository+<GetRemoteItemsAsync>d__21.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()",
        "Curse.Radium.Addons.BaseAddonServiceDatabase`3+<UpdateCacheAsync>d__14.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Runtime.CompilerServices.TaskAwaiter.GetResult()",
        "Curse.Radium.Addons.BaseAddonServiceDatabase`3+<InitializeAsync>d__8.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Runtime.CompilerServices.TaskAwaiter.GetResult()",
        "Curse.AppService.MinecraftAppService+<<InitializeAsync>b__22_0>d.MoveNext()"
      ]
    }
  }
},
{
  "timestamp": "2025-06-12T13:40:41.9304152-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Validating Minecraft installation"
},
{
  "timestamp": "2025-06-12T13:40:41.9304647-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "/home/avery/Documents/curseforge/minecraft/Install/minecraft-launcher found."
},
{
  "timestamp": "2025-06-12T13:40:42.4283106-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Validating Minecraft installation"
},
{
  "timestamp": "2025-06-12T13:40:42.4284224-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "/home/avery/Documents/curseforge/minecraft/Install/minecraft-launcher found."
},
{
  "timestamp": "2025-06-12T13:40:43.3976806-04:00",
  "level": "Info",
  "thread": 8,
  "memory": 0.0,
  "message": "Attempt #1 of 2 on InstalledAddonDatabase -> FetchAddonWithRetries"
},
{
  "timestamp": "2025-06-12T13:40:46.9219283-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Attempt #2 of 3 on InstalledAddonDatabase -> GetAddons"
},
{
  "timestamp": "2025-06-12T13:40:48.5205673-04:00",
  "level": "Info",
  "thread": 8,
  "memory": 0.0,
  "message": "Attempt #2 of 2 on InstalledAddonDatabase -> FetchAddonWithRetries"
},
{
  "timestamp": "2025-06-12T13:40:56.9458447-04:00",
  "level": "Info",
  "thread": 15,
  "memory": 0.0,
  "message": "Attempt #3 of 3 on InstalledAddonDatabase -> GetAddons"
},
{
  "timestamp": "2025-06-12T13:40:58.5472606-04:00",
  "level": "Error",
  "thread": 8,
  "memory": 0.0,
  "message": "Failed to get addon from service!",
  "error": {
    "message": "One or more errors occurred. (Failed getting project 1258033. Status Code: InternalServerError. Exception: System.Net.WebException: Web Service responded with failure status code: InternalServerError) (Failed getting project 1258033. Status Code: InternalServerError. Exception: System.Net.WebException: Web Service responded with failure status code: InternalServerError)",
    "type": "System.AggregateException",
    "stack": [
      "Curse.Radium.Utilities.Retry.Do[T](Func`1,IRetryStrategy,String,Func`2)",
      "Curse.Radium.Addons.ProjectRepository.FetchProjectWithRetries(Int32)",
      "Curse.Radium.Addons.ProjectRepository.UpdateProject(Int32)",
      "Curse.Radium.Addons.ProjectRepository.GetProjectById(Int32,Boolean)",
      "Curse.AppService.Minecraft.MinecraftInstanceMapper.Map(MinecraftGameInstance)",
      "System.Linq.Enumerable+SelectEnumerableIterator`2.MoveNext()",
      "System.Linq.Enumerable+WhereEnumerableIterator`1.ToArray()",
      "Curse.Plugin.API.Handlers.Minecraft.GameInstance.MinecraftGetInstancesRequestHandler.HandleRequest(MinecraftGetInstancesRequest,CancellationToken)",
      "Curse.Plugin.API.Handlers.AsyncApiRequestHandlerBase`1+<>c__DisplayClass7_0+<<Handle>b__0>d.MoveNext()",
      "System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine&)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[TStateMachine](TStateMachine&)",
      "Curse.Plugin.API.Handlers.AsyncApiRequestHandlerBase`1+<>c__DisplayClass7_0.<Handle>b__0()",
      "Curse.Plugin.RequestExecutor+<>c__DisplayClass9_0+<<ExecuteAsync>b__0>d.MoveNext()",
      "System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine&)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[TStateMachine](TStateMachine&)",
      "Curse.Plugin.RequestExecutor+<>c__DisplayClass9_0.<ExecuteAsync>b__0()",
      "System.Threading.Tasks.Task`1.InnerInvoke()",
      "System.Threading.Tasks.Task+<>c.<.cctor>b__281_0(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task&,Thread)",
      "System.Threading.Tasks.Task.ExecuteEntry()",
      "System.Threading.Tasks.TaskScheduler.TryExecuteTask(Task)",
      "System.Threading.Tasks.ConcurrentExclusiveSchedulerPair+ConcurrentExclusiveTaskScheduler.ExecuteTask(Task)",
      "System.Threading.Tasks.ConcurrentExclusiveSchedulerPair.ProcessExclusiveTasks()",
      "System.Threading.Tasks.ConcurrentExclusiveSchedulerPair+SchedulerWorkItem.System.Threading.IThreadPoolWorkItem.Execute()",
      "System.Threading.ThreadPoolWorkQueue.Dispatch()",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerDoWork(PortableThreadPool,Boolean&)",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerThreadStart()",
      "System.Threading.Thread+StartHelper.RunWorker()",
      "System.Threading.Thread+StartHelper.Run()",
      "System.Threading.Thread.StartCallback()"
    ],
    "innerError": {
      "message": "Failed getting project 1258033. Status Code: InternalServerError. Exception: System.Net.WebException: Web Service responded with failure status code: InternalServerError",
      "type": "Curse.Server.Communication.Exceptions.ServerRequestException",
      "stack": [
        "Curse.TechService.AddonServiceClient.HandleResponse[T](ServiceResponse`1,Func`1)",
        "Curse.TechService.AddonServiceClient.GetProject(Int32)",
        "Curse.Radium.Addons.ProjectRepository+<>c__DisplayClass31_0.<FetchProjectWithRetries>b__0()",
        "Curse.Radium.Utilities.Retry.Do[T](Func`1,IRetryStrategy,String,Func`2)"
      ]
    }
  }
},
{
  "timestamp": "2025-06-12T13:40:58.6770053-04:00",
  "level": "Info",
  "thread": 8,
  "memory": 0.0,
  "message": "Attempt #1 of 2 on InstalledAddonDatabase -> FetchAddonWithRetries"
},
{
  "timestamp": "2025-06-12T13:41:03.7025335-04:00",
  "level": "Info",
  "thread": 8,
  "memory": 0.0,
  "message": "Attempt #2 of 2 on InstalledAddonDatabase -> FetchAddonWithRetries"
},
{
  "timestamp": "2025-06-12T13:41:06.9679252-04:00",
  "level": "Error",
  "thread": 15,
  "memory": 0.0,
  "message": "UpdateCache Exception!",
  "error": {
    "message": "One or more errors occurred. (Connection refused (api.curseforge.com:443)) (Connection refused (api.curseforge.com:443)) (Connection refused (api.curseforge.com:443))",
    "type": "System.AggregateException",
    "stack": [
      "Curse.Radium.Utilities.Retry+<DoAsync>d__5`1.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "Curse.Radium.Addons.ProjectRepository+<UpdateProjectsAsync>d__22.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception)",
      "Curse.Radium.Utilities.Retry+<DoAsync>d__5`1.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task+DelayPromise.CompleteTimedOut()",
      "System.Threading.TimerQueueTimer.Fire(Boolean)",
      "System.Threading.TimerQueue.FireNextTimers()",
      "System.Threading.ThreadPoolWorkQueue.Dispatch()",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerDoWork(PortableThreadPool,Boolean&)",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerThreadStart()",
      "System.Threading.Thread+StartHelper.RunWorker()",
      "System.Threading.Thread+StartHelper.Run()",
      "System.Threading.Thread.StartCallback()"
    ],
    "innerError": {
      "message": "Connection refused (api.curseforge.com:443)",
      "type": "System.Net.Http.HttpRequestException",
      "stack": [
        "System.Net.Http.HttpConnectionPool+<ConnectToTcpHostAsync>d__104.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Threading.Tasks.ValueTask`1.get_Result()",
        "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
        "System.Net.Http.HttpConnectionPool+<ConnectAsync>d__103.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Threading.Tasks.ValueTask`1.get_Result()",
        "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
        "System.Net.Http.HttpConnectionPool+<CreateHttp11ConnectionAsync>d__105.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Threading.Tasks.ValueTask`1.get_Result()",
        "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
        "System.Net.Http.HttpConnectionPool+<AddHttp11ConnectionAsync>d__79.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter.GetResult()",
        "System.Threading.Tasks.TaskCompletionSourceWithCancellation`1+<WaitWithCancellationAsync>d__1.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Threading.Tasks.ValueTask`1.get_Result()",
        "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
        "System.Net.Http.HttpConnectionPool+<SendWithVersionDetectionAndRetryAsync>d__89.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Threading.Tasks.ValueTask`1.get_Result()",
        "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
        "System.Net.Http.RedirectHandler+<SendAsync>d__4.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Threading.Tasks.ValueTask`1.get_Result()",
        "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
        "System.Net.Http.DecompressionHandler+<SendAsync>d__16.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter.GetResult()",
        "System.Net.Http.HttpClient+<<SendAsync>g__Core|83_0>d.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()",
        "Curse.Server.Communication.BaseWebServiceClient+<PostAsync>d__15`1.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()",
        "Curse.TechService.AddonServiceClient+<GetProjectsAsync>d__11.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()",
        "Curse.Radium.Addons.ProjectRepository+<>c__DisplayClass22_0+<<UpdateProjectsAsync>b__0>d.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Runtime.CompilerServices.TaskAwaiter.GetResult()",
        "Curse.Radium.Utilities.Retry+<>c__DisplayClass4_0+<<DoAsync>b__0>d.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()",
        "Curse.Radium.Utilities.Retry+<DoAsync>d__5`1.MoveNext()"
      ],
      "innerError": {
        "message": "Connection refused",
        "type": "System.Net.Sockets.SocketException",
        "stack": [
          "System.Net.Sockets.Socket+AwaitableSocketAsyncEventArgs.ThrowException(SocketError,CancellationToken)",
          "System.Net.Sockets.Socket+AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource.GetResult(Int16)",
          "System.Threading.Tasks.ValueTask.ThrowIfCompletedUnsuccessfully()",
          "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable+ConfiguredValueTaskAwaiter.GetResult()",
          "System.Net.Sockets.Socket+<<ConnectAsync>g__WaitForConnectWithCancellation|285_0>d.MoveNext()",
          "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
          "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
          "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
          "System.Threading.Tasks.ValueTask.ThrowIfCompletedUnsuccessfully()",
          "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable+ConfiguredValueTaskAwaiter.GetResult()",
          "System.Net.Http.HttpConnectionPool+<ConnectToTcpHostAsync>d__104.MoveNext()"
        ]
      }
    }
  }
},
{
  "timestamp": "2025-06-12T13:41:13.7380596-04:00",
  "level": "Error",
  "thread": 8,
  "memory": 0.0,
  "message": "Failed to get addon from service!",
  "error": {
    "message": "One or more errors occurred. (Failed getting project 1258033. Status Code: InternalServerError. Exception: System.Net.WebException: Web Service responded with failure status code: InternalServerError) (Failed getting project 1258033. Status Code: InternalServerError. Exception: System.Net.WebException: Web Service responded with failure status code: InternalServerError)",
    "type": "System.AggregateException",
    "stack": [
      "Curse.Radium.Utilities.Retry.Do[T](Func`1,IRetryStrategy,String,Func`2)",
      "Curse.Radium.Addons.ProjectRepository.FetchProjectWithRetries(Int32)",
      "Curse.Radium.Addons.ProjectRepository.UpdateProject(Int32)",
      "Curse.Radium.Addons.ProjectRepository.GetProjectById(Int32,Boolean)",
      "Curse.Games.Minecraft.MinecraftGameInstance.get_Addon()",
      "Curse.Games.Minecraft.MinecraftGameInstance.get_Author()",
      "Curse.AppService.Minecraft.MinecraftInstanceMapper.Map(MinecraftGameInstance)",
      "System.Linq.Enumerable+SelectEnumerableIterator`2.MoveNext()",
      "System.Linq.Enumerable+WhereEnumerableIterator`1.ToArray()",
      "Curse.Plugin.API.Handlers.Minecraft.GameInstance.MinecraftGetInstancesRequestHandler.HandleRequest(MinecraftGetInstancesRequest,CancellationToken)",
      "Curse.Plugin.API.Handlers.AsyncApiRequestHandlerBase`1+<>c__DisplayClass7_0+<<Handle>b__0>d.MoveNext()",
      "System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine&)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[TStateMachine](TStateMachine&)",
      "Curse.Plugin.API.Handlers.AsyncApiRequestHandlerBase`1+<>c__DisplayClass7_0.<Handle>b__0()",
      "Curse.Plugin.RequestExecutor+<>c__DisplayClass9_0+<<ExecuteAsync>b__0>d.MoveNext()",
      "System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine&)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[TStateMachine](TStateMachine&)",
      "Curse.Plugin.RequestExecutor+<>c__DisplayClass9_0.<ExecuteAsync>b__0()",
      "System.Threading.Tasks.Task`1.InnerInvoke()",
      "System.Threading.Tasks.Task+<>c.<.cctor>b__281_0(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task&,Thread)",
      "System.Threading.Tasks.Task.ExecuteEntry()",
      "System.Threading.Tasks.TaskScheduler.TryExecuteTask(Task)",
      "System.Threading.Tasks.ConcurrentExclusiveSchedulerPair+ConcurrentExclusiveTaskScheduler.ExecuteTask(Task)",
      "System.Threading.Tasks.ConcurrentExclusiveSchedulerPair.ProcessExclusiveTasks()",
      "System.Threading.Tasks.ConcurrentExclusiveSchedulerPair+SchedulerWorkItem.System.Threading.IThreadPoolWorkItem.Execute()",
      "System.Threading.ThreadPoolWorkQueue.Dispatch()",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerDoWork(PortableThreadPool,Boolean&)",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerThreadStart()",
      "System.Threading.Thread+StartHelper.RunWorker()",
      "System.Threading.Thread+StartHelper.Run()",
      "System.Threading.Thread.StartCallback()"
    ],
    "innerError": {
      "message": "Failed getting project 1258033. Status Code: InternalServerError. Exception: System.Net.WebException: Web Service responded with failure status code: InternalServerError",
      "type": "Curse.Server.Communication.Exceptions.ServerRequestException",
      "stack": [
        "Curse.TechService.AddonServiceClient.HandleResponse[T](ServiceResponse`1,Func`1)",
        "Curse.TechService.AddonServiceClient.GetProject(Int32)",
        "Curse.Radium.Addons.ProjectRepository+<>c__DisplayClass31_0.<FetchProjectWithRetries>b__0()",
        "Curse.Radium.Utilities.Retry.Do[T](Func`1,IRetryStrategy,String,Func`2)"
      ]
    }
  }
},
{
  "timestamp": "2025-06-12T13:41:13.7405077-04:00",
  "level": "Info",
  "thread": 8,
  "memory": 0.0,
  "message": "Attempt #1 of 2 on InstalledAddonDatabase -> FetchAddonWithRetries"
},
{
  "timestamp": "2025-06-12T13:41:18.7598213-04:00",
  "level": "Info",
  "thread": 8,
  "memory": 0.0,
  "message": "Attempt #2 of 2 on InstalledAddonDatabase -> FetchAddonWithRetries"
},
{
  "timestamp": "2025-06-12T13:41:28.7798798-04:00",
  "level": "Error",
  "thread": 8,
  "memory": 0.0,
  "message": "Failed to get addon from service!",
  "error": {
    "message": "One or more errors occurred. (Failed getting project 1258033. Status Code: InternalServerError. Exception: System.Net.WebException: Web Service responded with failure status code: InternalServerError) (Failed getting project 1258033. Status Code: InternalServerError. Exception: System.Net.WebException: Web Service responded with failure status code: InternalServerError)",
    "type": "System.AggregateException",
    "stack": [
      "Curse.Radium.Utilities.Retry.Do[T](Func`1,IRetryStrategy,String,Func`2)",
      "Curse.Radium.Addons.ProjectRepository.FetchProjectWithRetries(Int32)",
      "Curse.Radium.Addons.ProjectRepository.UpdateProject(Int32)",
      "Curse.Radium.Addons.ProjectRepository.GetProjectById(Int32,Boolean)",
      "Curse.Games.Minecraft.MinecraftGameInstance.get_Addon()",
      "Curse.Games.Minecraft.MinecraftGameInstance.get_UpdateFileID()",
      "Curse.Games.Minecraft.MinecraftGameInstance.get_UpdateAvailable()",
      "Curse.AppService.Minecraft.MinecraftInstanceMapper.Map(MinecraftGameInstance)",
      "System.Linq.Enumerable+SelectEnumerableIterator`2.MoveNext()",
      "System.Linq.Enumerable+WhereEnumerableIterator`1.ToArray()",
      "Curse.Plugin.API.Handlers.Minecraft.GameInstance.MinecraftGetInstancesRequestHandler.HandleRequest(MinecraftGetInstancesRequest,CancellationToken)",
      "Curse.Plugin.API.Handlers.AsyncApiRequestHandlerBase`1+<>c__DisplayClass7_0+<<Handle>b__0>d.MoveNext()",
      "System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine&)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[TStateMachine](TStateMachine&)",
      "Curse.Plugin.API.Handlers.AsyncApiRequestHandlerBase`1+<>c__DisplayClass7_0.<Handle>b__0()",
      "Curse.Plugin.RequestExecutor+<>c__DisplayClass9_0+<<ExecuteAsync>b__0>d.MoveNext()",
      "System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine&)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[TStateMachine](TStateMachine&)",
      "Curse.Plugin.RequestExecutor+<>c__DisplayClass9_0.<ExecuteAsync>b__0()",
      "System.Threading.Tasks.Task`1.InnerInvoke()",
      "System.Threading.Tasks.Task+<>c.<.cctor>b__281_0(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task&,Thread)",
      "System.Threading.Tasks.Task.ExecuteEntry()",
      "System.Threading.Tasks.TaskScheduler.TryExecuteTask(Task)",
      "System.Threading.Tasks.ConcurrentExclusiveSchedulerPair+ConcurrentExclusiveTaskScheduler.ExecuteTask(Task)",
      "System.Threading.Tasks.ConcurrentExclusiveSchedulerPair.ProcessExclusiveTasks()",
      "System.Threading.Tasks.ConcurrentExclusiveSchedulerPair+SchedulerWorkItem.System.Threading.IThreadPoolWorkItem.Execute()",
      "System.Threading.ThreadPoolWorkQueue.Dispatch()",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerDoWork(PortableThreadPool,Boolean&)",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerThreadStart()",
      "System.Threading.Thread+StartHelper.RunWorker()",
      "System.Threading.Thread+StartHelper.Run()",
      "System.Threading.Thread.StartCallback()"
    ],
    "innerError": {
      "message": "Failed getting project 1258033. Status Code: InternalServerError. Exception: System.Net.WebException: Web Service responded with failure status code: InternalServerError",
      "type": "Curse.Server.Communication.Exceptions.ServerRequestException",
      "stack": [
        "Curse.TechService.AddonServiceClient.HandleResponse[T](ServiceResponse`1,Func`1)",
        "Curse.TechService.AddonServiceClient.GetProject(Int32)",
        "Curse.Radium.Addons.ProjectRepository+<>c__DisplayClass31_0.<FetchProjectWithRetries>b__0()",
        "Curse.Radium.Utilities.Retry.Do[T](Func`1,IRetryStrategy,String,Func`2)"
      ]
    }
  }
},
{
  "timestamp": "2025-06-12T13:41:28.9100421-04:00",
  "level": "Info",
  "thread": 8,
  "memory": 0.0,
  "message": "Validating Minecraft installation"
},
{
  "timestamp": "2025-06-12T13:41:28.910087-04:00",
  "level": "Info",
  "thread": 8,
  "memory": 0.0,
  "message": "/home/avery/Documents/curseforge/minecraft/Install/minecraft-launcher found."
},
{
  "timestamp": "2025-06-12T13:41:29.2700476-04:00",
  "level": "Info",
  "thread": 18,
  "memory": 0.0,
  "message": "Attempt #1 of 2 on InstalledAddonDatabase -> FetchAddonWithRetries"
},
{
  "timestamp": "2025-06-12T13:41:34.2940076-04:00",
  "level": "Info",
  "thread": 18,
  "memory": 0.0,
  "message": "Attempt #2 of 2 on InstalledAddonDatabase -> FetchAddonWithRetries"
},
{
  "timestamp": "2025-06-12T13:41:44.3148326-04:00",
  "level": "Error",
  "thread": 18,
  "memory": 0.0,
  "message": "Failed to get addon from service!",
  "error": {
    "message": "One or more errors occurred. (Failed getting project 1258033. Status Code: InternalServerError. Exception: System.Net.WebException: Web Service responded with failure status code: InternalServerError) (Failed getting project 1258033. Status Code: InternalServerError. Exception: System.Net.WebException: Web Service responded with failure status code: InternalServerError)",
    "type": "System.AggregateException",
    "stack": [
      "Curse.Radium.Utilities.Retry.Do[T](Func`1,IRetryStrategy,String,Func`2)",
      "Curse.Radium.Addons.ProjectRepository.FetchProjectWithRetries(Int32)",
      "Curse.Radium.Addons.ProjectRepository.UpdateProject(Int32)",
      "Curse.Radium.Addons.ProjectRepository.GetProjectById(Int32,Boolean)",
      "Curse.AppService.Minecraft.MinecraftInstanceMapper.Map(MinecraftGameInstance)",
      "System.Linq.Enumerable+SelectEnumerableIterator`2.MoveNext()",
      "System.Linq.Enumerable+WhereEnumerableIterator`1.ToArray()",
      "Curse.Plugin.API.Handlers.Minecraft.GameInstance.MinecraftGetInstancesRequestHandler.HandleRequest(MinecraftGetInstancesRequest,CancellationToken)",
      "Curse.Plugin.API.Handlers.AsyncApiRequestHandlerBase`1+<>c__DisplayClass7_0+<<Handle>b__0>d.MoveNext()",
      "System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine&)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[TStateMachine](TStateMachine&)",
      "Curse.Plugin.API.Handlers.AsyncApiRequestHandlerBase`1+<>c__DisplayClass7_0.<Handle>b__0()",
      "Curse.Plugin.RequestExecutor+<>c__DisplayClass9_0+<<ExecuteAsync>b__0>d.MoveNext()",
      "System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine&)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[TStateMachine](TStateMachine&)",
      "Curse.Plugin.RequestExecutor+<>c__DisplayClass9_0.<ExecuteAsync>b__0()",
      "System.Threading.Tasks.Task`1.InnerInvoke()",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task&,Thread)",
      "System.Threading.Tasks.Task.ExecuteEntry()",
      "System.Threading.Tasks.TaskScheduler.TryExecuteTask(Task)",
      "System.Threading.Tasks.ConcurrentExclusiveSchedulerPair+ConcurrentExclusiveTaskScheduler.ExecuteTask(Task)",
      "System.Threading.Tasks.ConcurrentExclusiveSchedulerPair.ProcessExclusiveTasks()",
      "System.Threading.Tasks.ConcurrentExclusiveSchedulerPair+SchedulerWorkItem.System.Threading.IThreadPoolWorkItem.Execute()",
      "System.Threading.ThreadPoolWorkQueue.Dispatch()",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerDoWork(PortableThreadPool,Boolean&)",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerThreadStart()",
      "System.Threading.Thread+StartHelper.RunWorker()",
      "System.Threading.Thread+StartHelper.Run()",
      "System.Threading.Thread.StartCallback()"
    ],
    "innerError": {
      "message": "Failed getting project 1258033. Status Code: InternalServerError. Exception: System.Net.WebException: Web Service responded with failure status code: InternalServerError",
      "type": "Curse.Server.Communication.Exceptions.ServerRequestException",
      "stack": [
        "Curse.TechService.AddonServiceClient.HandleResponse[T](ServiceResponse`1,Func`1)",
        "Curse.TechService.AddonServiceClient.GetProject(Int32)",
        "Curse.Radium.Addons.ProjectRepository+<>c__DisplayClass31_0.<FetchProjectWithRetries>b__0()",
        "Curse.Radium.Utilities.Retry.Do[T](Func`1,IRetryStrategy,String,Func`2)"
      ]
    }
  }
},
{
  "timestamp": "2025-06-12T13:41:44.3201912-04:00",
  "level": "Info",
  "thread": 18,
  "memory": 0.0,
  "message": "Attempt #1 of 2 on InstalledAddonDatabase -> FetchAddonWithRetries"
},
{
  "timestamp": "2025-06-12T13:41:49.3453816-04:00",
  "level": "Info",
  "thread": 18,
  "memory": 0.0,
  "message": "Attempt #2 of 2 on InstalledAddonDatabase -> FetchAddonWithRetries"
},
{
  "timestamp": "2025-06-12T13:41:59.3660229-04:00",
  "level": "Error",
  "thread": 18,
  "memory": 0.0,
  "message": "Failed to get addon from service!",
  "error": {
    "message": "One or more errors occurred. (Failed getting project 1258033. Status Code: InternalServerError. Exception: System.Net.WebException: Web Service responded with failure status code: InternalServerError) (Failed getting project 1258033. Status Code: InternalServerError. Exception: System.Net.WebException: Web Service responded with failure status code: InternalServerError)",
    "type": "System.AggregateException",
    "stack": [
      "Curse.Radium.Utilities.Retry.Do[T](Func`1,IRetryStrategy,String,Func`2)",
      "Curse.Radium.Addons.ProjectRepository.FetchProjectWithRetries(Int32)",
      "Curse.Radium.Addons.ProjectRepository.UpdateProject(Int32)",
      "Curse.Radium.Addons.ProjectRepository.GetProjectById(Int32,Boolean)",
      "Curse.Games.Minecraft.MinecraftGameInstance.get_Addon()",
      "Curse.Games.Minecraft.MinecraftGameInstance.get_Author()",
      "Curse.AppService.Minecraft.MinecraftInstanceMapper.Map(MinecraftGameInstance)",
      "System.Linq.Enumerable+SelectEnumerableIterator`2.MoveNext()",
      "System.Linq.Enumerable+WhereEnumerableIterator`1.ToArray()",
      "Curse.Plugin.API.Handlers.Minecraft.GameInstance.MinecraftGetInstancesRequestHandler.HandleRequest(MinecraftGetInstancesRequest,CancellationToken)",
      "Curse.Plugin.API.Handlers.AsyncApiRequestHandlerBase`1+<>c__DisplayClass7_0+<<Handle>b__0>d.MoveNext()",
      "System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine&)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[TStateMachine](TStateMachine&)",
      "Curse.Plugin.API.Handlers.AsyncApiRequestHandlerBase`1+<>c__DisplayClass7_0.<Handle>b__0()",
      "Curse.Plugin.RequestExecutor+<>c__DisplayClass9_0+<<ExecuteAsync>b__0>d.MoveNext()",
      "System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine&)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[TStateMachine](TStateMachine&)",
      "Curse.Plugin.RequestExecutor+<>c__DisplayClass9_0.<ExecuteAsync>b__0()",
      "System.Threading.Tasks.Task`1.InnerInvoke()",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task&,Thread)",
      "System.Threading.Tasks.Task.ExecuteEntry()",
      "System.Threading.Tasks.TaskScheduler.TryExecuteTask(Task)",
      "System.Threading.Tasks.ConcurrentExclusiveSchedulerPair+ConcurrentExclusiveTaskScheduler.ExecuteTask(Task)",
      "System.Threading.Tasks.ConcurrentExclusiveSchedulerPair.ProcessExclusiveTasks()",
      "System.Threading.Tasks.ConcurrentExclusiveSchedulerPair+SchedulerWorkItem.System.Threading.IThreadPoolWorkItem.Execute()",
      "System.Threading.ThreadPoolWorkQueue.Dispatch()",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerDoWork(PortableThreadPool,Boolean&)",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerThreadStart()",
      "System.Threading.Thread+StartHelper.RunWorker()",
      "System.Threading.Thread+StartHelper.Run()",
      "System.Threading.Thread.StartCallback()"
    ],
    "innerError": {
      "message": "Failed getting project 1258033. Status Code: InternalServerError. Exception: System.Net.WebException: Web Service responded with failure status code: InternalServerError",
      "type": "Curse.Server.Communication.Exceptions.ServerRequestException",
      "stack": [
        "Curse.TechService.AddonServiceClient.HandleResponse[T](ServiceResponse`1,Func`1)",
        "Curse.TechService.AddonServiceClient.GetProject(Int32)",
        "Curse.Radium.Addons.ProjectRepository+<>c__DisplayClass31_0.<FetchProjectWithRetries>b__0()",
        "Curse.Radium.Utilities.Retry.Do[T](Func`1,IRetryStrategy,String,Func`2)"
      ]
    }
  }
},
{
  "timestamp": "2025-06-12T13:41:59.3672581-04:00",
  "level": "Info",
  "thread": 18,
  "memory": 0.0,
  "message": "Attempt #1 of 2 on InstalledAddonDatabase -> FetchAddonWithRetries"
},
{
  "timestamp": "2025-06-12T13:42:04.3888903-04:00",
  "level": "Info",
  "thread": 18,
  "memory": 0.0,
  "message": "Attempt #2 of 2 on InstalledAddonDatabase -> FetchAddonWithRetries"
},
{
  "timestamp": "2025-06-12T13:42:14.4888013-04:00",
  "level": "Error",
  "thread": 18,
  "memory": 0.0,
  "message": "Failed to get addon from service!",
  "error": {
    "message": "One or more errors occurred. (Failed getting project 1258033. Status Code: InternalServerError. Exception: System.Net.WebException: Web Service responded with failure status code: InternalServerError) (Failed getting project 1258033. Status Code: InternalServerError. Exception: System.Net.WebException: Web Service responded with failure status code: InternalServerError)",
    "type": "System.AggregateException",
    "stack": [
      "Curse.Radium.Utilities.Retry.Do[T](Func`1,IRetryStrategy,String,Func`2)",
      "Curse.Radium.Addons.ProjectRepository.FetchProjectWithRetries(Int32)",
      "Curse.Radium.Addons.ProjectRepository.UpdateProject(Int32)",
      "Curse.Radium.Addons.ProjectRepository.GetProjectById(Int32,Boolean)",
      "Curse.Games.Minecraft.MinecraftGameInstance.get_Addon()",
      "Curse.Games.Minecraft.MinecraftGameInstance.get_UpdateFileID()",
      "Curse.Games.Minecraft.MinecraftGameInstance.get_UpdateAvailable()",
      "Curse.AppService.Minecraft.MinecraftInstanceMapper.Map(MinecraftGameInstance)",
      "System.Linq.Enumerable+SelectEnumerableIterator`2.MoveNext()",
      "System.Linq.Enumerable+WhereEnumerableIterator`1.ToArray()",
      "Curse.Plugin.API.Handlers.Minecraft.GameInstance.MinecraftGetInstancesRequestHandler.HandleRequest(MinecraftGetInstancesRequest,CancellationToken)",
      "Curse.Plugin.API.Handlers.AsyncApiRequestHandlerBase`1+<>c__DisplayClass7_0+<<Handle>b__0>d.MoveNext()",
      "System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine&)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[TStateMachine](TStateMachine&)",
      "Curse.Plugin.API.Handlers.AsyncApiRequestHandlerBase`1+<>c__DisplayClass7_0.<Handle>b__0()",
      "Curse.Plugin.RequestExecutor+<>c__DisplayClass9_0+<<ExecuteAsync>b__0>d.MoveNext()",
      "System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine&)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[TStateMachine](TStateMachine&)",
      "Curse.Plugin.RequestExecutor+<>c__DisplayClass9_0.<ExecuteAsync>b__0()",
      "System.Threading.Tasks.Task`1.InnerInvoke()",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task&,Thread)",
      "System.Threading.Tasks.Task.ExecuteEntry()",
      "System.Threading.Tasks.TaskScheduler.TryExecuteTask(Task)",
      "System.Threading.Tasks.ConcurrentExclusiveSchedulerPair+ConcurrentExclusiveTaskScheduler.ExecuteTask(Task)",
      "System.Threading.Tasks.ConcurrentExclusiveSchedulerPair.ProcessExclusiveTasks()",
      "System.Threading.Tasks.ConcurrentExclusiveSchedulerPair+SchedulerWorkItem.System.Threading.IThreadPoolWorkItem.Execute()",
      "System.Threading.ThreadPoolWorkQueue.Dispatch()",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerDoWork(PortableThreadPool,Boolean&)",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerThreadStart()",
      "System.Threading.Thread+StartHelper.RunWorker()",
      "System.Threading.Thread+StartHelper.Run()",
      "System.Threading.Thread.StartCallback()"
    ],
    "innerError": {
      "message": "Failed getting project 1258033. Status Code: InternalServerError. Exception: System.Net.WebException: Web Service responded with failure status code: InternalServerError",
      "type": "Curse.Server.Communication.Exceptions.ServerRequestException",
      "stack": [
        "Curse.TechService.AddonServiceClient.HandleResponse[T](ServiceResponse`1,Func`1)",
        "Curse.TechService.AddonServiceClient.GetProject(Int32)",
        "Curse.Radium.Addons.ProjectRepository+<>c__DisplayClass31_0.<FetchProjectWithRetries>b__0()",
        "Curse.Radium.Utilities.Retry.Do[T](Func`1,IRetryStrategy,String,Func`2)"
      ]
    }
  }
},
{
  "timestamp": "2025-06-12T13:42:14.4914625-04:00",
  "level": "Info",
  "thread": 18,
  "memory": 0.0,
  "message": "Attempt #1 of 2 on InstalledAddonDatabase -> FetchAddonWithRetries"
},
{
  "timestamp": "2025-06-12T13:42:19.5142114-04:00",
  "level": "Info",
  "thread": 18,
  "memory": 0.0,
  "message": "Attempt #2 of 2 on InstalledAddonDatabase -> FetchAddonWithRetries"
},
{
  "timestamp": "2025-06-12T13:42:29.5344639-04:00",
  "level": "Error",
  "thread": 18,
  "memory": 0.0,
  "message": "Failed to get addon from service!",
  "error": {
    "message": "One or more errors occurred. (Failed getting project 1258033. Status Code: InternalServerError. Exception: System.Net.WebException: Web Service responded with failure status code: InternalServerError) (Failed getting project 1258033. Status Code: InternalServerError. Exception: System.Net.WebException: Web Service responded with failure status code: InternalServerError)",
    "type": "System.AggregateException",
    "stack": [
      "Curse.Radium.Utilities.Retry.Do[T](Func`1,IRetryStrategy,String,Func`2)",
      "Curse.Radium.Addons.ProjectRepository.FetchProjectWithRetries(Int32)",
      "Curse.Radium.Addons.ProjectRepository.UpdateProject(Int32)",
      "Curse.Radium.Addons.ProjectRepository.GetProjectById(Int32,Boolean)",
      "Curse.AppService.Minecraft.MinecraftInstanceMapper.Map(MinecraftGameInstance)",
      "System.Linq.Enumerable+SelectEnumerableIterator`2.MoveNext()",
      "System.Linq.Enumerable+WhereEnumerableIterator`1.ToArray()",
      "Curse.Plugin.API.Handlers.Minecraft.GameInstance.MinecraftGetInstancesRequestHandler.HandleRequest(MinecraftGetInstancesRequest,CancellationToken)",
      "Curse.Plugin.API.Handlers.AsyncApiRequestHandlerBase`1+<>c__DisplayClass7_0+<<Handle>b__0>d.MoveNext()",
      "System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine&)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[TStateMachine](TStateMachine&)",
      "Curse.Plugin.API.Handlers.AsyncApiRequestHandlerBase`1+<>c__DisplayClass7_0.<Handle>b__0()",
      "Curse.Plugin.RequestExecutor+<>c__DisplayClass9_0+<<ExecuteAsync>b__0>d.MoveNext()",
      "System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine&)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[TStateMachine](TStateMachine&)",
      "Curse.Plugin.RequestExecutor+<>c__DisplayClass9_0.<ExecuteAsync>b__0()",
      "System.Threading.Tasks.Task`1.InnerInvoke()",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task&,Thread)",
      "System.Threading.Tasks.Task.ExecuteEntry()",
      "System.Threading.Tasks.ConcurrentExclusiveSchedulerPair+ConcurrentExclusiveTaskScheduler.ExecuteTask(Task)",
      "System.Threading.Tasks.ConcurrentExclusiveSchedulerPair.ProcessExclusiveTasks()",
      "System.Threading.Tasks.ConcurrentExclusiveSchedulerPair+SchedulerWorkItem.System.Threading.IThreadPoolWorkItem.Execute()",
      "System.Threading.ThreadPoolWorkQueue.Dispatch()",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerDoWork(PortableThreadPool,Boolean&)",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerThreadStart()",
      "System.Threading.Thread+StartHelper.RunWorker()",
      "System.Threading.Thread+StartHelper.Run()",
      "System.Threading.Thread.StartCallback()"
    ],
    "innerError": {
      "message": "Failed getting project 1258033. Status Code: InternalServerError. Exception: System.Net.WebException: Web Service responded with failure status code: InternalServerError",
      "type": "Curse.Server.Communication.Exceptions.ServerRequestException",
      "stack": [
        "Curse.TechService.AddonServiceClient.HandleResponse[T](ServiceResponse`1,Func`1)",
        "Curse.TechService.AddonServiceClient.GetProject(Int32)",
        "Curse.Radium.Addons.ProjectRepository+<>c__DisplayClass31_0.<FetchProjectWithRetries>b__0()",
        "Curse.Radium.Utilities.Retry.Do[T](Func`1,IRetryStrategy,String,Func`2)"
      ]
    }
  }
},
{
  "timestamp": "2025-06-12T13:42:29.5364246-04:00",
  "level": "Info",
  "thread": 18,
  "memory": 0.0,
  "message": "Attempt #1 of 2 on InstalledAddonDatabase -> FetchAddonWithRetries"
},
{
  "timestamp": "2025-06-12T13:42:34.5557175-04:00",
  "level": "Info",
  "thread": 18,
  "memory": 0.0,
  "message": "Attempt #2 of 2 on InstalledAddonDatabase -> FetchAddonWithRetries"
},
{
  "timestamp": "2025-06-12T13:42:44.5752805-04:00",
  "level": "Error",
  "thread": 18,
  "memory": 0.0,
  "message": "Failed to get addon from service!",
  "error": {
    "message": "One or more errors occurred. (Failed getting project 1258033. Status Code: InternalServerError. Exception: System.Net.WebException: Web Service responded with failure status code: InternalServerError) (Failed getting project 1258033. Status Code: InternalServerError. Exception: System.Net.WebException: Web Service responded with failure status code: InternalServerError)",
    "type": "System.AggregateException",
    "stack": [
      "Curse.Radium.Utilities.Retry.Do[T](Func`1,IRetryStrategy,String,Func`2)",
      "Curse.Radium.Addons.ProjectRepository.FetchProjectWithRetries(Int32)",
      "Curse.Radium.Addons.ProjectRepository.UpdateProject(Int32)",
      "Curse.Radium.Addons.ProjectRepository.GetProjectById(Int32,Boolean)",
      "Curse.Games.Minecraft.MinecraftGameInstance.get_Addon()",
      "Curse.Games.Minecraft.MinecraftGameInstance.get_Author()",
      "Curse.AppService.Minecraft.MinecraftInstanceMapper.Map(MinecraftGameInstance)",
      "System.Linq.Enumerable+SelectEnumerableIterator`2.MoveNext()",
      "System.Linq.Enumerable+WhereEnumerableIterator`1.ToArray()",
      "Curse.Plugin.API.Handlers.Minecraft.GameInstance.MinecraftGetInstancesRequestHandler.HandleRequest(MinecraftGetInstancesRequest,CancellationToken)",
      "Curse.Plugin.API.Handlers.AsyncApiRequestHandlerBase`1+<>c__DisplayClass7_0+<<Handle>b__0>d.MoveNext()",
      "System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine&)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[TStateMachine](TStateMachine&)",
      "Curse.Plugin.API.Handlers.AsyncApiRequestHandlerBase`1+<>c__DisplayClass7_0.<Handle>b__0()",
      "Curse.Plugin.RequestExecutor+<>c__DisplayClass9_0+<<ExecuteAsync>b__0>d.MoveNext()",
      "System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine&)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[TStateMachine](TStateMachine&)",
      "Curse.Plugin.RequestExecutor+<>c__DisplayClass9_0.<ExecuteAsync>b__0()",
      "System.Threading.Tasks.Task`1.InnerInvoke()",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task&,Thread)",
      "System.Threading.Tasks.Task.ExecuteEntry()",
      "System.Threading.Tasks.ConcurrentExclusiveSchedulerPair+ConcurrentExclusiveTaskScheduler.ExecuteTask(Task)",
      "System.Threading.Tasks.ConcurrentExclusiveSchedulerPair.ProcessExclusiveTasks()",
      "System.Threading.Tasks.ConcurrentExclusiveSchedulerPair+SchedulerWorkItem.System.Threading.IThreadPoolWorkItem.Execute()",
      "System.Threading.ThreadPoolWorkQueue.Dispatch()",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerDoWork(PortableThreadPool,Boolean&)",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerThreadStart()",
      "System.Threading.Thread+StartHelper.RunWorker()",
      "System.Threading.Thread+StartHelper.Run()",
      "System.Threading.Thread.StartCallback()"
    ],
    "innerError": {
      "message": "Failed getting project 1258033. Status Code: InternalServerError. Exception: System.Net.WebException: Web Service responded with failure status code: InternalServerError",
      "type": "Curse.Server.Communication.Exceptions.ServerRequestException",
      "stack": [
        "Curse.TechService.AddonServiceClient.HandleResponse[T](ServiceResponse`1,Func`1)",
        "Curse.TechService.AddonServiceClient.GetProject(Int32)",
        "Curse.Radium.Addons.ProjectRepository+<>c__DisplayClass31_0.<FetchProjectWithRetries>b__0()",
        "Curse.Radium.Utilities.Retry.Do[T](Func`1,IRetryStrategy,String,Func`2)"
      ]
    }
  }
},
{
  "timestamp": "2025-06-12T13:42:44.575915-04:00",
  "level": "Info",
  "thread": 18,
  "memory": 0.0,
  "message": "Attempt #1 of 2 on InstalledAddonDatabase -> FetchAddonWithRetries"
},
{
  "timestamp": "2025-06-12T13:42:49.5925953-04:00",
  "level": "Info",
  "thread": 18,
  "memory": 0.0,
  "message": "Attempt #2 of 2 on InstalledAddonDatabase -> FetchAddonWithRetries"
},
{
  "timestamp": "2025-06-12T13:42:53.7067457-04:00",
  "level": "Info",
  "thread": 11,
  "memory": 0.0,
  "message": "Network availability has changed. Checking connection"
},
{
  "timestamp": "2025-06-12T13:42:53.7128535-04:00",
  "level": "Info",
  "thread": 11,
  "memory": 0.0,
  "message": "Network connected and internet available"
},
{
  "timestamp": "2025-06-12T13:42:59.6148334-04:00",
  "level": "Error",
  "thread": 18,
  "memory": 0.0,
  "message": "Failed to get addon from service!",
  "error": {
    "message": "One or more errors occurred. (Failed getting project 1258033. Status Code: InternalServerError. Exception: System.Net.WebException: Web Service responded with failure status code: InternalServerError) (Failed getting project 1258033. Status Code: InternalServerError. Exception: System.Net.WebException: Web Service responded with failure status code: InternalServerError)",
    "type": "System.AggregateException",
    "stack": [
      "Curse.Radium.Utilities.Retry.Do[T](Func`1,IRetryStrategy,String,Func`2)",
      "Curse.Radium.Addons.ProjectRepository.FetchProjectWithRetries(Int32)",
      "Curse.Radium.Addons.ProjectRepository.UpdateProject(Int32)",
      "Curse.Radium.Addons.ProjectRepository.GetProjectById(Int32,Boolean)",
      "Curse.Games.Minecraft.MinecraftGameInstance.get_Addon()",
      "Curse.Games.Minecraft.MinecraftGameInstance.get_UpdateFileID()",
      "Curse.Games.Minecraft.MinecraftGameInstance.get_UpdateAvailable()",
      "Curse.AppService.Minecraft.MinecraftInstanceMapper.Map(MinecraftGameInstance)",
      "System.Linq.Enumerable+SelectEnumerableIterator`2.MoveNext()",
      "System.Linq.Enumerable+WhereEnumerableIterator`1.ToArray()",
      "Curse.Plugin.API.Handlers.Minecraft.GameInstance.MinecraftGetInstancesRequestHandler.HandleRequest(MinecraftGetInstancesRequest,CancellationToken)",
      "Curse.Plugin.API.Handlers.AsyncApiRequestHandlerBase`1+<>c__DisplayClass7_0+<<Handle>b__0>d.MoveNext()",
      "System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine&)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[TStateMachine](TStateMachine&)",
      "Curse.Plugin.API.Handlers.AsyncApiRequestHandlerBase`1+<>c__DisplayClass7_0.<Handle>b__0()",
      "Curse.Plugin.RequestExecutor+<>c__DisplayClass9_0+<<ExecuteAsync>b__0>d.MoveNext()",
      "System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine&)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[TStateMachine](TStateMachine&)",
      "Curse.Plugin.RequestExecutor+<>c__DisplayClass9_0.<ExecuteAsync>b__0()",
      "System.Threading.Tasks.Task`1.InnerInvoke()",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task&,Thread)",
      "System.Threading.Tasks.Task.ExecuteEntry()",
      "System.Threading.Tasks.ConcurrentExclusiveSchedulerPair+ConcurrentExclusiveTaskScheduler.ExecuteTask(Task)",
      "System.Threading.Tasks.ConcurrentExclusiveSchedulerPair.ProcessExclusiveTasks()",
      "System.Threading.Tasks.ConcurrentExclusiveSchedulerPair+SchedulerWorkItem.System.Threading.IThreadPoolWorkItem.Execute()",
      "System.Threading.ThreadPoolWorkQueue.Dispatch()",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerDoWork(PortableThreadPool,Boolean&)",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerThreadStart()",
      "System.Threading.Thread+StartHelper.RunWorker()",
      "System.Threading.Thread+StartHelper.Run()",
      "System.Threading.Thread.StartCallback()"
    ],
    "innerError": {
      "message": "Failed getting project 1258033. Status Code: InternalServerError. Exception: System.Net.WebException: Web Service responded with failure status code: InternalServerError",
      "type": "Curse.Server.Communication.Exceptions.ServerRequestException",
      "stack": [
        "Curse.TechService.AddonServiceClient.HandleResponse[T](ServiceResponse`1,Func`1)",
        "Curse.TechService.AddonServiceClient.GetProject(Int32)",
        "Curse.Radium.Addons.ProjectRepository+<>c__DisplayClass31_0.<FetchProjectWithRetries>b__0()",
        "Curse.Radium.Utilities.Retry.Do[T](Func`1,IRetryStrategy,String,Func`2)"
      ]
    }
  }
},
{
  "timestamp": "2025-06-12T13:42:59.6314962-04:00",
  "level": "Info",
  "thread": 18,
  "memory": 0.0,
  "message": "Attempt #1 of 2 on InstalledAddonDatabase -> FetchAddonWithRetries"
},
{
  "timestamp": "2025-06-12T13:43:00.6959595-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Network address has changed. Checking connection"
},
{
  "timestamp": "2025-06-12T13:43:00.6984588-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Network status changed: Disconnected"
},
{
  "timestamp": "2025-06-12T13:43:00.7926943-04:00",
  "level": "Info",
  "thread": 15,
  "memory": 0.0,
  "message": "Network availability has changed. Checking connection"
},
{
  "timestamp": "2025-06-12T13:43:01.0214377-04:00",
  "level": "Info",
  "thread": 15,
  "memory": 0.0,
  "message": "Network availability has changed. Checking connection"
},
{
  "timestamp": "2025-06-12T13:43:01.3624741-04:00",
  "level": "Info",
  "thread": 15,
  "memory": 0.0,
  "message": "Network availability has changed. Checking connection"
},
{
  "timestamp": "2025-06-12T13:43:01.551526-04:00",
  "level": "Info",
  "thread": 11,
  "memory": 0.0,
  "message": "Network availability has changed. Checking connection"
},
{
  "timestamp": "2025-06-12T13:43:01.5976553-04:00",
  "level": "Info",
  "thread": 11,
  "memory": 0.0,
  "message": "Network connected, but internet not available. Waiting for connection"
},
{
  "timestamp": "2025-06-12T13:43:01.5977097-04:00",
  "level": "Info",
  "thread": 11,
  "memory": 0.0,
  "message": "Network status changed: Reconnecting"
},
{
  "timestamp": "2025-06-12T13:43:01.7076437-04:00",
  "level": "Info",
  "thread": 11,
  "memory": 0.0,
  "message": "Network address has changed. Checking connection"
},
{
  "timestamp": "2025-06-12T13:43:01.7231028-04:00",
  "level": "Info",
  "thread": 11,
  "memory": 0.0,
  "message": "Network connected, but internet not available. Waiting for connection"
},
{
  "timestamp": "2025-06-12T13:43:01.8750166-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Network availability has changed. Checking connection"
},
{
  "timestamp": "2025-06-12T13:43:02.2150241-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Network connected and internet available"
},
{
  "timestamp": "2025-06-12T13:43:02.2150867-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Network status changed: Connected"
},
{
  "timestamp": "2025-06-12T13:43:04.6567904-04:00",
  "level": "Info",
  "thread": 18,
  "memory": 0.0,
  "message": "Attempt #2 of 2 on InstalledAddonDatabase -> FetchAddonWithRetries"
},
{
  "timestamp": "2025-06-12T13:43:05.8306381-04:00",
  "level": "Info",
  "thread": 11,
  "memory": 0.0,
  "message": "Network availability has changed. Checking connection"
},
{
  "timestamp": "2025-06-12T13:43:05.8362799-04:00",
  "level": "Info",
  "thread": 11,
  "memory": 0.0,
  "message": "Network connected and internet available"
},
{
  "timestamp": "2025-06-12T13:43:06.7121311-04:00",
  "level": "Info",
  "thread": 18,
  "memory": 0.0,
  "message": "Curse Mods Agent is shutting down"
},
{
  "timestamp": "2025-06-12T13:43:06.7122755-04:00",
  "level": "Info",
  "thread": 18,
  "memory": 0.0,
  "message": "Shutting down periodic tasks"
},
{
  "timestamp": "2025-06-12T13:43:06.7124334-04:00",
  "level": "Info",
  "thread": 18,
  "memory": 0.0,
  "message": "Disabling periodic timers"
},
{
  "timestamp": "2025-06-12T13:43:06.7131424-04:00",
  "level": "Info",
  "thread": 18,
  "memory": 0.0,
  "message": "[InstallTaskManager] Install Manager Shutting Down"
},
{
  "timestamp": "2025-06-12T13:43:06.7134051-04:00",
  "level": "Info",
  "thread": 18,
  "memory": 0.0,
  "message": "[InstallTaskManager] Install Manager Shutting Down"
},
{
  "timestamp": "2025-06-12T13:43:06.7134107-04:00",
  "level": "Info",
  "thread": 18,
  "memory": 0.0,
  "message": "[InstallTaskManager] Install Manager Shutting Down"
},
{
  "timestamp": "2025-06-12T13:43:06.7136028-04:00",
  "level": "Info",
  "thread": 18,
  "memory": 0.0,
  "message": "Shutdown game instances."
},
{
  "timestamp": "2025-06-12T13:43:06.7160296-04:00",
  "level": "Info",
  "thread": 18,
  "memory": 0.0,
  "message": "[MinecraftGameInstance] GameInstance.SaveInstancesToDisk: Saving MinecraftGameInstance DB with 3 items"
},
{
  "timestamp": "2025-06-12T13:43:08.5605663-04:00",
  "level": "Info",
  "thread": 15,
  "memory": 0.0,
  "message": "Network availability has changed. Checking connection"
},
{
  "timestamp": "2025-06-12T13:43:08.5660792-04:00",
  "level": "Info",
  "thread": 15,
  "memory": 0.0,
  "message": "Network connected and internet available"
},
{
  "timestamp": "2025-06-12T13:43:13.6815839-04:00",
  "level": "Info",
  "thread": 15,
  "memory": 0.0,
  "message": "Network availability has changed. Checking connection"
},
{
  "timestamp": "2025-06-12T13:43:13.7202981-04:00",
  "level": "Info",
  "thread": 15,
  "memory": 0.0,
  "message": "Network connected and internet available"
},
{
  "timestamp": "2025-06-12T13:43:23.3845003-04:00",
  "level": "Info",
  "thread": 14,
  "memory": 0.0,
  "message": "Network availability has changed. Checking connection"
},
{
  "timestamp": "2025-06-12T13:43:23.3902084-04:00",
  "level": "Info",
  "thread": 14,
  "memory": 0.0,
  "message": "Network connected and internet available"
},
{
  "timestamp": "2025-06-12T13:43:26.7837867-04:00",
  "level": "Info",
  "thread": 14,
  "memory": 0.0,
  "message": "Network availability has changed. Checking connection"
},
{
  "timestamp": "2025-06-12T13:43:26.7874425-04:00",
  "level": "Info",
  "thread": 14,
  "memory": 0.0,
  "message": "Network connected and internet available"
},
{
  "timestamp": "2025-06-12T13:43:30.6839917-04:00",
  "level": "Info",
  "thread": 15,
  "memory": 0.0,
  "message": "Network availability has changed. Checking connection"
},
{
  "timestamp": "2025-06-12T13:43:30.6904386-04:00",
  "level": "Info",
  "thread": 15,
  "memory": 0.0,
  "message": "Network connected and internet available"
},
{
  "timestamp": "2025-06-12T13:43:34.4653376-04:00",
  "level": "Info",
  "thread": 14,
  "memory": 0.0,
  "message": "Network availability has changed. Checking connection"
},
{
  "timestamp": "2025-06-12T13:43:34.4711328-04:00",
  "level": "Info",
  "thread": 14,
  "memory": 0.0,
  "message": "Network connected and internet available"
},
{
  "timestamp": "2025-06-12T13:43:39.606116-04:00",
  "level": "Info",
  "thread": 14,
  "memory": 0.0,
  "message": "Network availability has changed. Checking connection"
},
{
  "timestamp": "2025-06-12T13:43:39.6264635-04:00",
  "level": "Info",
  "thread": 14,
  "memory": 0.0,
  "message": "Network connected and internet available"
},
{
  "timestamp": "2025-06-12T13:43:43.4901089-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Network availability has changed. Checking connection"
},
{
  "timestamp": "2025-06-12T13:43:43.4949747-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Network connected and internet available"
},
{
  "timestamp": "2025-06-12T13:43:55.6485783-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Network availability has changed. Checking connection"
},
{
  "timestamp": "2025-06-12T13:43:55.6544583-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Network connected and internet available"
},
{
  "timestamp": "2025-06-12T13:43:58.4099132-04:00",
  "level": "Info",
  "thread": 18,
  "memory": 0.0,
  "message": "[MinecraftGameInstance] GameInstance.SaveInstancesToDisk: Saved MinecraftGameInstance DB, which took 51693ms"
},
{
  "timestamp": "2025-06-12T13:43:58.4113234-04:00",
  "level": "Info",
  "thread": 18,
  "memory": 0.0,
  "message": "[AddonGameInstance] GameInstance.SaveInstancesToDisk: Saving GameInstance DB with 0 items"
},
{
  "timestamp": "2025-06-12T13:43:58.5580299-04:00",
  "level": "Info",
  "thread": 18,
  "memory": 0.0,
  "message": "[AddonGameInstance] GameInstance.SaveInstancesToDisk: Saved GameInstance DB, which took 146ms"
},
{
  "timestamp": "2025-06-12T13:43:58.5773445-04:00",
  "level": "Info",
  "thread": 18,
  "memory": 0.0,
  "message": "Clearing Discord presence."
},
{
  "timestamp": "2025-06-12T13:43:58.5806557-04:00",
  "level": "Info",
  "thread": 18,
  "memory": 0.0,
  "message": "Discord RPC client disposed successfully."
},
{
  "timestamp": "2025-06-12T13:43:58.9402037-04:00",
  "level": "Info",
  "thread": 14,
  "memory": 0.0,
  "message": "Network availability has changed. Checking connection"
},
{
  "timestamp": "2025-06-12T13:43:58.9455955-04:00",
  "level": "Info",
  "thread": 14,
  "memory": 0.0,
  "message": "Network connected and internet available"
},
{
  "timestamp": "2025-06-12T13:43:59.2528373-04:00",
  "level": "Info",
  "thread": 14,
  "memory": 0.0,
  "message": "Web Socket session disconnected"
},
{
  "timestamp": "2025-06-12T13:43:59.2584612-04:00",
  "level": "Warn",
  "thread": 14,
  "memory": 0.0,
  "message": "Client disconnected from server"
},
{
  "timestamp": "2025-06-12T13:43:59.466248-04:00",
  "level": "Warn",
  "thread": 5,
  "memory": 0.0,
  "message": "Parent process is no longer active. Mods Agent will shutdown."
},
{
  "timestamp": "2025-06-12T13:43:59.4667742-04:00",
  "level": "Warn",
  "thread": 1,
  "memory": 0.0,
  "message": "Parent process exit. Shutting down..."
},
{
  "timestamp": "2025-06-12T13:43:59.4865315-04:00",
  "level": "Info",
  "thread": 2,
  "memory": 0.0,
  "message": "Shutting down the logger"
},
{
  "timestamp": "2025-06-12T13:43:59.5413136-04:00",
  "level": "Info",
  "thread": 2,
  "memory": 0.0,
  "message": "Logging worker finished"
}
]}
